// tspbbot-for-rq/lib/bot/Bot.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Bot = void 0;
var lru_cache_1 = __importDefault(require("lru-cache"));
var EventHandler_1 = require("./EventHandler");
var Msg_1 = require("../util/Msg");
var IdGenerator_1 = require("../util/IdGenerator");
var onebot = __importStar(require("../proto/onebot_frame"));
var Frame = onebot.Frame;
var FrameType = onebot.Frame_FrameType;
var Bot = /** @class */ (function () {
    function Bot(botId, session) {
        var _this = this;
        this.waitingFrames = new lru_cache_1.default({
            max: 500,
            maxAge: 30 * 1000,
            dispose: function (key, value) {
                value.reject("waiting timeout");
            }
        });
        this.botId = botId;
        this.session = session;
        this.session.on('message', function (data) { return __awaiter(_this, void 0, void 0, function () {
            var frame, e_1;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        frame = Frame.decode(data);
                        _a.label = 1;
                    case 1:
                        _a.trys.push([1, 3, , 4]);
                        return [4 /*yield*/, this.handleFrame(frame)];
                    case 2:
                        _a.sent();
                        return [3 /*break*/, 4];
                    case 3:
                        e_1 = _a.sent();
                        console.log("failed to handle frame");
                        console.error(e_1);
                        return [3 /*break*/, 4];
                    case 4: return [2 /*return*/];
                }
            });
        }); });
    }
    Bot.prototype.handleFrame = function (frame) {
        return __awaiter(this, void 0, void 0, function () {
            var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, waitingFrame;
            return __generator(this, function (_o) {
                switch (_o.label) {
                    case 0:
                        _a = !!frame.privateMessageEvent;
                        if (!_a) return [3 /*break*/, 2];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handlePrivateMessage(this, frame.privateMessageEvent)];
                    case 1:
                        _a = (_o.sent());
                        _o.label = 2;
                    case 2:
                        _a;
                        _b = !!frame.groupMessageEvent;
                        if (!_b) return [3 /*break*/, 4];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupMessage(this, frame.groupMessageEvent)];
                    case 3:
                        _b = (_o.sent());
                        _o.label = 4;
                    case 4:
                        _b;
                        _c = !!frame.groupUploadNoticeEvent;
                        if (!_c) return [3 /*break*/, 6];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupUploadNotice(this, frame.groupUploadNoticeEvent)];
                    case 5:
                        _c = (_o.sent());
                        _o.label = 6;
                    case 6:
                        _c;
                        _d = !!frame.groupAdminNoticeEvent;
                        if (!_d) return [3 /*break*/, 8];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupAdminNotice(this, frame.groupAdminNoticeEvent)];
                    case 7:
                        _d = (_o.sent());
                        _o.label = 8;
                    case 8:
                        _d;
                        _e = !!frame.groupDecreaseNoticeEvent;
                        if (!_e) return [3 /*break*/, 10];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupDecreaseNotice(this, frame.groupDecreaseNoticeEvent)];
                    case 9:
                        _e = (_o.sent());
                        _o.label = 10;
                    case 10:
                        _e;
                        _f = !!frame.groupIncreaseNoticeEvent;
                        if (!_f) return [3 /*break*/, 12];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupIncreaseNotice(this, frame.groupIncreaseNoticeEvent)];
                    case 11:
                        _f = (_o.sent());
                        _o.label = 12;
                    case 12:
                        _f;
                        _g = !!frame.groupBanNoticeEvent;
                        if (!_g) return [3 /*break*/, 14];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupBanNotice(this, frame.groupBanNoticeEvent)];
                    case 13:
                        _g = (_o.sent());
                        _o.label = 14;
                    case 14:
                        _g;
                        _h = !!frame.friendAddNoticeEvent;
                        if (!_h) return [3 /*break*/, 16];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleFriendAddNotice(this, frame.friendAddNoticeEvent)];
                    case 15:
                        _h = (_o.sent());
                        _o.label = 16;
                    case 16:
                        _h;
                        _j = !!frame.groupRecallNoticeEvent;
                        if (!_j) return [3 /*break*/, 18];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupRecallNotice(this, frame.groupRecallNoticeEvent)];
                    case 17:
                        _j = (_o.sent());
                        _o.label = 18;
                    case 18:
                        _j;
                        _k = !!frame.friendRecallNoticeEvent;
                        if (!_k) return [3 /*break*/, 20];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleFriendRecallNotice(this, frame.friendRecallNoticeEvent)];
                    case 19:
                        _k = (_o.sent());
                        _o.label = 20;
                    case 20:
                        _k;
                        _l = !!frame.friendRequestEvent;
                        if (!_l) return [3 /*break*/, 22];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleFriendRequest(this, frame.friendRequestEvent)];
                    case 21:
                        _l = (_o.sent());
                        _o.label = 22;
                    case 22:
                        _l;
                        _m = !!frame.groupRequestEvent;
                        if (!_m) return [3 /*break*/, 24];
                        return [4 /*yield*/, EventHandler_1.EventHandler.handleGroupRequest(this, frame.groupRequestEvent)];
                    case 23:
                        _m = (_o.sent());
                        _o.label = 24;
                    case 24:
                        _m;
                        if ((frame.frameType || 0) > 200) {
                            waitingFrame = this.waitingFrames.get(frame.echo);
                            if (!!waitingFrame) {
                                waitingFrame.resolve(frame);
                            }
                            this.waitingFrames.del(frame.echo);
                        }
                        return [2 /*return*/];
                }
            });
        });
    };
    Bot.prototype.sendFrame = function (frame) {
        var sendingData = Frame.encode(frame).finish();
        this.session.send(sendingData);
    };
    Bot.prototype.sendFrameAndWait = function (frame) {
        return __awaiter(this, void 0, void 0, function () {
            var _this = this;
            return __generator(this, function (_a) {
                frame.botId = this.botId;
                frame.echo = IdGenerator_1.IdGenerator.generateStrId();
                frame.ok = true;
                this.sendFrame(frame);
                return [2 /*return*/, new Promise(function (resolve, reject) {
                        var waitingFrame = {
                            resolve: resolve,
                            reject: reject,
                            echo: frame.echo,
                        };
                        _this.waitingFrames.set(frame.echo, waitingFrame);
                    })];
            });
        });
    };
    Bot.prototype.sendPrivateMessage = function (userId, msg, autoEscape) {
        if (autoEscape === void 0) { autoEscape = true; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (typeof msg == 'string') {
                            msg = Msg_1.Msg.builder().text(msg);
                        }
                        return [4 /*yield*/, this.sendFrameAndWait({
                                frameType: FrameType.TSendPrivateMsgReq,
                                sendPrivateMsgReq: {
                                    userId: Number(userId),
                                    message: msg.messageList,
                                    autoEscape: autoEscape,
                                },
                                botId: 0,
                                echo: "",
                                ok: false,
                                extra: {},
                                privateMessageEvent: undefined,
                                groupMessageEvent: undefined,
                                groupUploadNoticeEvent: undefined,
                                groupAdminNoticeEvent: undefined,
                                groupDecreaseNoticeEvent: undefined,
                                groupIncreaseNoticeEvent: undefined,
                                groupBanNoticeEvent: undefined,
                                friendAddNoticeEvent: undefined,
                                groupRecallNoticeEvent: undefined,
                                friendRecallNoticeEvent: undefined,
                                friendRequestEvent: undefined,
                                groupRequestEvent: undefined,
                                sendGroupMsgReq: undefined,
                                sendMsgReq: undefined,
                                deleteMsgReq: undefined,
                                getMsgReq: undefined,
                                getForwardMsgReq: undefined,
                                sendLikeReq: undefined,
                                setGroupKickReq: undefined,
                                setGroupBanReq: undefined,
                                setGroupAnonymousBanReq: undefined,
                                setGroupWholeBanReq: undefined,
                                setGroupAdminReq: undefined,
                                setGroupAnonymousReq: undefined,
                                setGroupCardReq: undefined,
                                setGroupNameReq: undefined,
                                setGroupLeaveReq: undefined,
                                setGroupSpecialTitleReq: undefined,
                                setFriendAddRequestReq: undefined,
                                setGroupAddRequestReq: undefined,
                                getLoginInfoReq: undefined,
                                getStrangerInfoReq: undefined,
                                getFriendListReq: undefined,
                                getGroupInfoReq: undefined,
                                getGroupListReq: undefined,
                                getGroupMemberInfoReq: undefined,
                                getGroupMemberListReq: undefined,
                                getGroupHonorInfoReq: undefined,
                                getCookiesReq: undefined,
                                getCsrfTokenReq: undefined,
                                getCredentialsReq: undefined,
                                getRecordReq: undefined,
                                getImageReq: undefined,
                                canSendImageReq: undefined,
                                canSendRecordReq: undefined,
                                getStatusReq: undefined,
                                getVersionInfoReq: undefined,
                                setRestartReq: undefined,
                                cleanCacheReq: undefined,
                                setGroupSignInReq: undefined,
                                sendPrivateMsgResp: undefined,
                                sendGroupMsgResp: undefined,
                                sendMsgResp: undefined,
                                deleteMsgResp: undefined,
                                getMsgResp: undefined,
                                getForwardMsgResp: undefined,
                                sendLikeResp: undefined,
                                setGroupKickResp: undefined,
                                setGroupBanResp: undefined,
                                setGroupAnonymousBanResp: undefined,
                                setGroupWholeBanResp: undefined,
                                setGroupAdminResp: undefined,
                                setGroupAnonymousResp: undefined,
                                setGroupCardResp: undefined,
                                setGroupNameResp: undefined,
                                setGroupLeaveResp: undefined,
                                setGroupSpecialTitleResp: undefined,
                                setFriendAddRequestResp: undefined,
                                setGroupAddRequestResp: undefined,
                                getLoginInfoResp: undefined,
                                getStrangerInfoResp: undefined,
                                getFriendListResp: undefined,
                                getGroupInfoResp: undefined,
                                getGroupListResp: undefined,
                                getGroupMemberInfoResp: undefined,
                                getGroupMemberListResp: undefined,
                                getGroupHonorInfoResp: undefined,
                                getCookiesResp: undefined,
                                getCsrfTokenResp: undefined,
                                getCredentialsResp: undefined,
                                getRecordResp: undefined,
                                getImageResp: undefined,
                                canSendImageResp: undefined,
                                canSendRecordResp: undefined,
                                getStatusResp: undefined,
                                getVersionInfoResp: undefined,
                                setRestartResp: undefined,
                                cleanCacheResp: undefined,
                                setGroupSignInResp: undefined
                            }).then(function (resp) { return resp.sendPrivateMsgResp || null; })
                                .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.sendGroupMessage = function (groupId, msg, autoEscape) {
        if (autoEscape === void 0) { autoEscape = true; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (typeof msg == 'string') {
                            msg = Msg_1.Msg.builder().text(msg);
                        }
                        return [4 /*yield*/, this.sendFrameAndWait({
                                frameType: FrameType.TSendGroupMsgReq,
                                sendGroupMsgReq: {
                                    groupId: Number(groupId),
                                    message: msg.messageList,
                                    autoEscape: autoEscape,
                                },
                                botId: 0,
                                echo: "",
                                ok: false,
                                extra: {},
                                privateMessageEvent: undefined,
                                groupMessageEvent: undefined,
                                groupUploadNoticeEvent: undefined,
                                groupAdminNoticeEvent: undefined,
                                groupDecreaseNoticeEvent: undefined,
                                groupIncreaseNoticeEvent: undefined,
                                groupBanNoticeEvent: undefined,
                                friendAddNoticeEvent: undefined,
                                groupRecallNoticeEvent: undefined,
                                friendRecallNoticeEvent: undefined,
                                friendRequestEvent: undefined,
                                groupRequestEvent: undefined,
                                sendPrivateMsgReq: undefined,
                                sendMsgReq: undefined,
                                deleteMsgReq: undefined,
                                getMsgReq: undefined,
                                getForwardMsgReq: undefined,
                                sendLikeReq: undefined,
                                setGroupKickReq: undefined,
                                setGroupBanReq: undefined,
                                setGroupAnonymousBanReq: undefined,
                                setGroupWholeBanReq: undefined,
                                setGroupAdminReq: undefined,
                                setGroupAnonymousReq: undefined,
                                setGroupCardReq: undefined,
                                setGroupNameReq: undefined,
                                setGroupLeaveReq: undefined,
                                setGroupSpecialTitleReq: undefined,
                                setFriendAddRequestReq: undefined,
                                setGroupAddRequestReq: undefined,
                                getLoginInfoReq: undefined,
                                getStrangerInfoReq: undefined,
                                getFriendListReq: undefined,
                                getGroupInfoReq: undefined,
                                getGroupListReq: undefined,
                                getGroupMemberInfoReq: undefined,
                                getGroupMemberListReq: undefined,
                                getGroupHonorInfoReq: undefined,
                                getCookiesReq: undefined,
                                getCsrfTokenReq: undefined,
                                getCredentialsReq: undefined,
                                getRecordReq: undefined,
                                getImageReq: undefined,
                                canSendImageReq: undefined,
                                canSendRecordReq: undefined,
                                getStatusReq: undefined,
                                getVersionInfoReq: undefined,
                                setRestartReq: undefined,
                                cleanCacheReq: undefined,
                                setGroupSignInReq: undefined,
                                sendPrivateMsgResp: undefined,
                                sendGroupMsgResp: undefined,
                                sendMsgResp: undefined,
                                deleteMsgResp: undefined,
                                getMsgResp: undefined,
                                getForwardMsgResp: undefined,
                                sendLikeResp: undefined,
                                setGroupKickResp: undefined,
                                setGroupBanResp: undefined,
                                setGroupAnonymousBanResp: undefined,
                                setGroupWholeBanResp: undefined,
                                setGroupAdminResp: undefined,
                                setGroupAnonymousResp: undefined,
                                setGroupCardResp: undefined,
                                setGroupNameResp: undefined,
                                setGroupLeaveResp: undefined,
                                setGroupSpecialTitleResp: undefined,
                                setFriendAddRequestResp: undefined,
                                setGroupAddRequestResp: undefined,
                                getLoginInfoResp: undefined,
                                getStrangerInfoResp: undefined,
                                getFriendListResp: undefined,
                                getGroupInfoResp: undefined,
                                getGroupListResp: undefined,
                                getGroupMemberInfoResp: undefined,
                                getGroupMemberListResp: undefined,
                                getGroupHonorInfoResp: undefined,
                                getCookiesResp: undefined,
                                getCsrfTokenResp: undefined,
                                getCredentialsResp: undefined,
                                getRecordResp: undefined,
                                getImageResp: undefined,
                                canSendImageResp: undefined,
                                canSendRecordResp: undefined,
                                getStatusResp: undefined,
                                getVersionInfoResp: undefined,
                                setRestartResp: undefined,
                                cleanCacheResp: undefined,
                                setGroupSignInResp: undefined
                            }).then(function (resp) { return resp.sendGroupMsgResp || null; })
                                .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.deleteMsg = function (messageId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TDeleteMsgReq,
                            deleteMsgReq: {
                                messageId: messageId,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.deleteMsgResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getMsg = function (messageId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetMsgReq,
                            getMsgReq: {
                                messageId: messageId,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getMsgResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setGroupKick = function (groupId, userId, rejectAddRequest) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetGroupKickReq,
                            setGroupKickReq: {
                                groupId: Number(groupId),
                                userId: Number(userId),
                                rejectAddRequest: rejectAddRequest,
                                userIds: [],
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setGroupKickResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setGroupBan = function (groupId, userId, duration) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetGroupBanReq,
                            setGroupBanReq: {
                                groupId: Number(groupId),
                                userId: Number(userId),
                                duration: duration,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setGroupBanResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setGroupWholeBan = function (groupId, enable) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetGroupWholeBanReq,
                            setGroupWholeBanReq: {
                                groupId: Number(groupId),
                                enable: enable,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setGroupWholeBanResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setGroupCard = function (groupId, userId, card) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetGroupCardReq,
                            setGroupCardReq: {
                                groupId: Number(groupId),
                                userId: Number(userId),
                                card: card,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setGroupCardResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setGroupLeave = function (groupId, isDismiss) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetGroupLeaveReq,
                            setGroupLeaveReq: {
                                groupId: Number(groupId),
                                isDismiss: isDismiss,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setGroupLeaveResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setGroupSpecialTitle = function (groupId, userId, specialTitle, duration) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetGroupSpecialTitleReq,
                            setGroupSpecialTitleReq: {
                                groupId: Number(groupId),
                                userId: Number(userId),
                                specialTitle: specialTitle,
                                duration: Number(duration),
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setGroupSpecialTitleResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setFriendAddRequest = function (flag, approve, remark) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetFriendAddRequestReq,
                            setFriendAddRequestReq: {
                                flag: flag,
                                approve: approve,
                                remark: remark
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setFriendAddRequestResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.setGroupAddRequest = function (flag, subType, type, approve, reason) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TSetGroupAddRequestReq,
                            setGroupAddRequestReq: {
                                flag: flag,
                                subType: subType,
                                type: type,
                                approve: approve,
                                reason: reason,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.setGroupAddRequestResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getLoginInfo = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetLoginInfoReq,
                            getLoginInfoReq: {},
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getLoginInfoResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getStrangerInfo = function (userId, noCache) {
        if (noCache === void 0) { noCache = true; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetStrangerInfoReq,
                            getStrangerInfoReq: {
                                userId: Number(userId),
                                noCache: noCache,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getStrangerInfoResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getFriendList = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetFriendListReq,
                            getFriendListReq: {},
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getFriendListResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getGroupList = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetGroupListReq,
                            getGroupListReq: {},
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getGroupListResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getGroupInfo = function (groupId, noCache) {
        if (noCache === void 0) { noCache = false; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetGroupInfoReq,
                            getGroupInfoReq: {
                                groupId: Number(groupId),
                                noCache: noCache,
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getGroupInfoResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getGroupMemberInfo = function (groupId, userId, noCache) {
        if (noCache === void 0) { noCache = false; }
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetGroupMemberInfoReq,
                            getGroupMemberInfoReq: {
                                groupId: Number(groupId),
                                userId: Number(userId),
                                noCache: noCache
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberListReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getGroupMemberInfoResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.prototype.getGroupMemberList = function (groupId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.sendFrameAndWait({
                            frameType: FrameType.TGetGroupMemberListReq,
                            getGroupMemberListReq: {
                                groupId: Number(groupId),
                            },
                            botId: 0,
                            echo: "",
                            ok: false,
                            extra: {},
                            privateMessageEvent: undefined,
                            groupMessageEvent: undefined,
                            groupUploadNoticeEvent: undefined,
                            groupAdminNoticeEvent: undefined,
                            groupDecreaseNoticeEvent: undefined,
                            groupIncreaseNoticeEvent: undefined,
                            groupBanNoticeEvent: undefined,
                            friendAddNoticeEvent: undefined,
                            groupRecallNoticeEvent: undefined,
                            friendRecallNoticeEvent: undefined,
                            friendRequestEvent: undefined,
                            groupRequestEvent: undefined,
                            sendPrivateMsgReq: undefined,
                            sendGroupMsgReq: undefined,
                            sendMsgReq: undefined,
                            deleteMsgReq: undefined,
                            getMsgReq: undefined,
                            getForwardMsgReq: undefined,
                            sendLikeReq: undefined,
                            setGroupKickReq: undefined,
                            setGroupBanReq: undefined,
                            setGroupAnonymousBanReq: undefined,
                            setGroupWholeBanReq: undefined,
                            setGroupAdminReq: undefined,
                            setGroupAnonymousReq: undefined,
                            setGroupCardReq: undefined,
                            setGroupNameReq: undefined,
                            setGroupLeaveReq: undefined,
                            setGroupSpecialTitleReq: undefined,
                            setFriendAddRequestReq: undefined,
                            setGroupAddRequestReq: undefined,
                            getLoginInfoReq: undefined,
                            getStrangerInfoReq: undefined,
                            getFriendListReq: undefined,
                            getGroupInfoReq: undefined,
                            getGroupListReq: undefined,
                            getGroupMemberInfoReq: undefined,
                            getGroupHonorInfoReq: undefined,
                            getCookiesReq: undefined,
                            getCsrfTokenReq: undefined,
                            getCredentialsReq: undefined,
                            getRecordReq: undefined,
                            getImageReq: undefined,
                            canSendImageReq: undefined,
                            canSendRecordReq: undefined,
                            getStatusReq: undefined,
                            getVersionInfoReq: undefined,
                            setRestartReq: undefined,
                            cleanCacheReq: undefined,
                            setGroupSignInReq: undefined,
                            sendPrivateMsgResp: undefined,
                            sendGroupMsgResp: undefined,
                            sendMsgResp: undefined,
                            deleteMsgResp: undefined,
                            getMsgResp: undefined,
                            getForwardMsgResp: undefined,
                            sendLikeResp: undefined,
                            setGroupKickResp: undefined,
                            setGroupBanResp: undefined,
                            setGroupAnonymousBanResp: undefined,
                            setGroupWholeBanResp: undefined,
                            setGroupAdminResp: undefined,
                            setGroupAnonymousResp: undefined,
                            setGroupCardResp: undefined,
                            setGroupNameResp: undefined,
                            setGroupLeaveResp: undefined,
                            setGroupSpecialTitleResp: undefined,
                            setFriendAddRequestResp: undefined,
                            setGroupAddRequestResp: undefined,
                            getLoginInfoResp: undefined,
                            getStrangerInfoResp: undefined,
                            getFriendListResp: undefined,
                            getGroupInfoResp: undefined,
                            getGroupListResp: undefined,
                            getGroupMemberInfoResp: undefined,
                            getGroupMemberListResp: undefined,
                            getGroupHonorInfoResp: undefined,
                            getCookiesResp: undefined,
                            getCsrfTokenResp: undefined,
                            getCredentialsResp: undefined,
                            getRecordResp: undefined,
                            getImageResp: undefined,
                            canSendImageResp: undefined,
                            canSendRecordResp: undefined,
                            getStatusResp: undefined,
                            getVersionInfoResp: undefined,
                            setRestartResp: undefined,
                            cleanCacheResp: undefined,
                            setGroupSignInResp: undefined
                        }).then(function (resp) { return resp.getGroupMemberListResp || null; })
                            .catch(function () { return null; })];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    Bot.bots = new Map();
    return Bot;
}());
exports.Bot = Bot;
