// tspbbot-for-rq/lib/proto/onebot_frame.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Frame_ExtraEntry = exports.Frame = exports.frame_FrameTypeToJSON = exports.frame_FrameTypeFromJSON = exports.Frame_FrameType = exports.protobufPackage = void 0;
/* eslint-disable */
var long_1 = __importDefault(require("long"));
var _m0 = __importStar(require("protobufjs/minimal"));
var onebot_event_1 = require("./onebot_event");
var onebot_api_1 = require("./onebot_api");
exports.protobufPackage = "onebot";
var Frame_FrameType;
(function (Frame_FrameType) {
    Frame_FrameType[Frame_FrameType["TUNKNOWN"] = 0] = "TUNKNOWN";
    Frame_FrameType[Frame_FrameType["TPrivateMessageEvent"] = 101] = "TPrivateMessageEvent";
    Frame_FrameType[Frame_FrameType["TGroupMessageEvent"] = 102] = "TGroupMessageEvent";
    Frame_FrameType[Frame_FrameType["TGroupUploadNoticeEvent"] = 103] = "TGroupUploadNoticeEvent";
    Frame_FrameType[Frame_FrameType["TGroupAdminNoticeEvent"] = 104] = "TGroupAdminNoticeEvent";
    Frame_FrameType[Frame_FrameType["TGroupDecreaseNoticeEvent"] = 105] = "TGroupDecreaseNoticeEvent";
    Frame_FrameType[Frame_FrameType["TGroupIncreaseNoticeEvent"] = 106] = "TGroupIncreaseNoticeEvent";
    Frame_FrameType[Frame_FrameType["TGroupBanNoticeEvent"] = 107] = "TGroupBanNoticeEvent";
    Frame_FrameType[Frame_FrameType["TFriendAddNoticeEvent"] = 108] = "TFriendAddNoticeEvent";
    Frame_FrameType[Frame_FrameType["TGroupRecallNoticeEvent"] = 109] = "TGroupRecallNoticeEvent";
    Frame_FrameType[Frame_FrameType["TFriendRecallNoticeEvent"] = 110] = "TFriendRecallNoticeEvent";
    Frame_FrameType[Frame_FrameType["TFriendRequestEvent"] = 111] = "TFriendRequestEvent";
    Frame_FrameType[Frame_FrameType["TGroupRequestEvent"] = 112] = "TGroupRequestEvent";
    Frame_FrameType[Frame_FrameType["TSendPrivateMsgReq"] = 201] = "TSendPrivateMsgReq";
    Frame_FrameType[Frame_FrameType["TSendGroupMsgReq"] = 202] = "TSendGroupMsgReq";
    Frame_FrameType[Frame_FrameType["TSendMsgReq"] = 203] = "TSendMsgReq";
    Frame_FrameType[Frame_FrameType["TDeleteMsgReq"] = 204] = "TDeleteMsgReq";
    Frame_FrameType[Frame_FrameType["TGetMsgReq"] = 205] = "TGetMsgReq";
    Frame_FrameType[Frame_FrameType["TGetForwardMsgReq"] = 206] = "TGetForwardMsgReq";
    Frame_FrameType[Frame_FrameType["TSendLikeReq"] = 207] = "TSendLikeReq";
    Frame_FrameType[Frame_FrameType["TSetGroupKickReq"] = 208] = "TSetGroupKickReq";
    Frame_FrameType[Frame_FrameType["TSetGroupBanReq"] = 209] = "TSetGroupBanReq";
    Frame_FrameType[Frame_FrameType["TSetGroupAnonymousReq"] = 210] = "TSetGroupAnonymousReq";
    Frame_FrameType[Frame_FrameType["TSetGroupWholeBanReq"] = 211] = "TSetGroupWholeBanReq";
    Frame_FrameType[Frame_FrameType["TSetGroupAdminReq"] = 212] = "TSetGroupAdminReq";
    Frame_FrameType[Frame_FrameType["TSetGroupAnonymousBanReq"] = 213] = "TSetGroupAnonymousBanReq";
    Frame_FrameType[Frame_FrameType["TSetGroupCardReq"] = 214] = "TSetGroupCardReq";
    Frame_FrameType[Frame_FrameType["TSetGroupNameReq"] = 215] = "TSetGroupNameReq";
    Frame_FrameType[Frame_FrameType["TSetGroupLeaveReq"] = 216] = "TSetGroupLeaveReq";
    Frame_FrameType[Frame_FrameType["TSetGroupSpecialTitleReq"] = 217] = "TSetGroupSpecialTitleReq";
    Frame_FrameType[Frame_FrameType["TSetFriendAddRequestReq"] = 218] = "TSetFriendAddRequestReq";
    Frame_FrameType[Frame_FrameType["TSetGroupAddRequestReq"] = 219] = "TSetGroupAddRequestReq";
    Frame_FrameType[Frame_FrameType["TGetLoginInfoReq"] = 220] = "TGetLoginInfoReq";
    Frame_FrameType[Frame_FrameType["TGetStrangerInfoReq"] = 221] = "TGetStrangerInfoReq";
    Frame_FrameType[Frame_FrameType["TGetFriendListReq"] = 222] = "TGetFriendListReq";
    Frame_FrameType[Frame_FrameType["TGetGroupInfoReq"] = 223] = "TGetGroupInfoReq";
    Frame_FrameType[Frame_FrameType["TGetGroupListReq"] = 224] = "TGetGroupListReq";
    Frame_FrameType[Frame_FrameType["TGetGroupMemberInfoReq"] = 225] = "TGetGroupMemberInfoReq";
    Frame_FrameType[Frame_FrameType["TGetGroupMemberListReq"] = 226] = "TGetGroupMemberListReq";
    Frame_FrameType[Frame_FrameType["TGetGroupHonorInfoReq"] = 227] = "TGetGroupHonorInfoReq";
    Frame_FrameType[Frame_FrameType["TGetCookiesReq"] = 228] = "TGetCookiesReq";
    Frame_FrameType[Frame_FrameType["TGetCsrfTokenReq"] = 229] = "TGetCsrfTokenReq";
    Frame_FrameType[Frame_FrameType["TGetCredentialsReq"] = 230] = "TGetCredentialsReq";
    Frame_FrameType[Frame_FrameType["TGetRecordReq"] = 231] = "TGetRecordReq";
    Frame_FrameType[Frame_FrameType["TGetImageReq"] = 232] = "TGetImageReq";
    Frame_FrameType[Frame_FrameType["TCanSendImageReq"] = 233] = "TCanSendImageReq";
    Frame_FrameType[Frame_FrameType["TCanSendRecordReq"] = 234] = "TCanSendRecordReq";
    Frame_FrameType[Frame_FrameType["TGetStatusReq"] = 235] = "TGetStatusReq";
    Frame_FrameType[Frame_FrameType["TGetVersionInfoReq"] = 236] = "TGetVersionInfoReq";
    Frame_FrameType[Frame_FrameType["TSetRestartReq"] = 237] = "TSetRestartReq";
    Frame_FrameType[Frame_FrameType["TCleanCacheReq"] = 238] = "TCleanCacheReq";
    Frame_FrameType[Frame_FrameType["TSetGroupSignInReq"] = 239] = "TSetGroupSignInReq";
    Frame_FrameType[Frame_FrameType["TSendPrivateMsgResp"] = 301] = "TSendPrivateMsgResp";
    Frame_FrameType[Frame_FrameType["TSendGroupMsgResp"] = 302] = "TSendGroupMsgResp";
    Frame_FrameType[Frame_FrameType["TSendMsgResp"] = 303] = "TSendMsgResp";
    Frame_FrameType[Frame_FrameType["TDeleteMsgResp"] = 304] = "TDeleteMsgResp";
    Frame_FrameType[Frame_FrameType["TGetMsgResp"] = 305] = "TGetMsgResp";
    Frame_FrameType[Frame_FrameType["TGetForwardMsgResp"] = 306] = "TGetForwardMsgResp";
    Frame_FrameType[Frame_FrameType["TSendLikeResp"] = 307] = "TSendLikeResp";
    Frame_FrameType[Frame_FrameType["TSetGroupKickResp"] = 308] = "TSetGroupKickResp";
    Frame_FrameType[Frame_FrameType["TSetGroupBanResp"] = 309] = "TSetGroupBanResp";
    Frame_FrameType[Frame_FrameType["TSetGroupAnonymousResp"] = 310] = "TSetGroupAnonymousResp";
    Frame_FrameType[Frame_FrameType["TSetGroupWholeBanResp"] = 311] = "TSetGroupWholeBanResp";
    Frame_FrameType[Frame_FrameType["TSetGroupAdminResp"] = 312] = "TSetGroupAdminResp";
    Frame_FrameType[Frame_FrameType["TSetGroupAnonymousBanResp"] = 313] = "TSetGroupAnonymousBanResp";
    Frame_FrameType[Frame_FrameType["TSetGroupCardResp"] = 314] = "TSetGroupCardResp";
    Frame_FrameType[Frame_FrameType["TSetGroupNameResp"] = 315] = "TSetGroupNameResp";
    Frame_FrameType[Frame_FrameType["TSetGroupLeaveResp"] = 316] = "TSetGroupLeaveResp";
    Frame_FrameType[Frame_FrameType["TSetGroupSpecialTitleResp"] = 317] = "TSetGroupSpecialTitleResp";
    Frame_FrameType[Frame_FrameType["TSetFriendAddRequestResp"] = 318] = "TSetFriendAddRequestResp";
    Frame_FrameType[Frame_FrameType["TSetGroupAddRequestResp"] = 319] = "TSetGroupAddRequestResp";
    Frame_FrameType[Frame_FrameType["TGetLoginInfoResp"] = 320] = "TGetLoginInfoResp";
    Frame_FrameType[Frame_FrameType["TGetStrangerInfoResp"] = 321] = "TGetStrangerInfoResp";
    Frame_FrameType[Frame_FrameType["TGetFriendListResp"] = 322] = "TGetFriendListResp";
    Frame_FrameType[Frame_FrameType["TGetGroupInfoResp"] = 323] = "TGetGroupInfoResp";
    Frame_FrameType[Frame_FrameType["TGetGroupListResp"] = 324] = "TGetGroupListResp";
    Frame_FrameType[Frame_FrameType["TGetGroupMemberInfoResp"] = 325] = "TGetGroupMemberInfoResp";
    Frame_FrameType[Frame_FrameType["TGetGroupMemberListResp"] = 326] = "TGetGroupMemberListResp";
    Frame_FrameType[Frame_FrameType["TGetGroupHonorInfoResp"] = 327] = "TGetGroupHonorInfoResp";
    Frame_FrameType[Frame_FrameType["TGetCookiesResp"] = 328] = "TGetCookiesResp";
    Frame_FrameType[Frame_FrameType["TGetCsrfTokenResp"] = 329] = "TGetCsrfTokenResp";
    Frame_FrameType[Frame_FrameType["TGetCredentialsResp"] = 330] = "TGetCredentialsResp";
    Frame_FrameType[Frame_FrameType["TGetRecordResp"] = 331] = "TGetRecordResp";
    Frame_FrameType[Frame_FrameType["TGetImageResp"] = 332] = "TGetImageResp";
    Frame_FrameType[Frame_FrameType["TCanSendImageResp"] = 333] = "TCanSendImageResp";
    Frame_FrameType[Frame_FrameType["TCanSendRecordResp"] = 334] = "TCanSendRecordResp";
    Frame_FrameType[Frame_FrameType["TGetStatusResp"] = 335] = "TGetStatusResp";
    Frame_FrameType[Frame_FrameType["TGetVersionInfoResp"] = 336] = "TGetVersionInfoResp";
    Frame_FrameType[Frame_FrameType["TSetRestartResp"] = 337] = "TSetRestartResp";
    Frame_FrameType[Frame_FrameType["TCleanCacheResp"] = 338] = "TCleanCacheResp";
    Frame_FrameType[Frame_FrameType["TSetGroupSignInResp"] = 339] = "TSetGroupSignInResp";
    Frame_FrameType[Frame_FrameType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(Frame_FrameType = exports.Frame_FrameType || (exports.Frame_FrameType = {}));
function frame_FrameTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "TUNKNOWN":
            return Frame_FrameType.TUNKNOWN;
        case 101:
        case "TPrivateMessageEvent":
            return Frame_FrameType.TPrivateMessageEvent;
        case 102:
        case "TGroupMessageEvent":
            return Frame_FrameType.TGroupMessageEvent;
        case 103:
        case "TGroupUploadNoticeEvent":
            return Frame_FrameType.TGroupUploadNoticeEvent;
        case 104:
        case "TGroupAdminNoticeEvent":
            return Frame_FrameType.TGroupAdminNoticeEvent;
        case 105:
        case "TGroupDecreaseNoticeEvent":
            return Frame_FrameType.TGroupDecreaseNoticeEvent;
        case 106:
        case "TGroupIncreaseNoticeEvent":
            return Frame_FrameType.TGroupIncreaseNoticeEvent;
        case 107:
        case "TGroupBanNoticeEvent":
            return Frame_FrameType.TGroupBanNoticeEvent;
        case 108:
        case "TFriendAddNoticeEvent":
            return Frame_FrameType.TFriendAddNoticeEvent;
        case 109:
        case "TGroupRecallNoticeEvent":
            return Frame_FrameType.TGroupRecallNoticeEvent;
        case 110:
        case "TFriendRecallNoticeEvent":
            return Frame_FrameType.TFriendRecallNoticeEvent;
        case 111:
        case "TFriendRequestEvent":
            return Frame_FrameType.TFriendRequestEvent;
        case 112:
        case "TGroupRequestEvent":
            return Frame_FrameType.TGroupRequestEvent;
        case 201:
        case "TSendPrivateMsgReq":
            return Frame_FrameType.TSendPrivateMsgReq;
        case 202:
        case "TSendGroupMsgReq":
            return Frame_FrameType.TSendGroupMsgReq;
        case 203:
        case "TSendMsgReq":
            return Frame_FrameType.TSendMsgReq;
        case 204:
        case "TDeleteMsgReq":
            return Frame_FrameType.TDeleteMsgReq;
        case 205:
        case "TGetMsgReq":
            return Frame_FrameType.TGetMsgReq;
        case 206:
        case "TGetForwardMsgReq":
            return Frame_FrameType.TGetForwardMsgReq;
        case 207:
        case "TSendLikeReq":
            return Frame_FrameType.TSendLikeReq;
        case 208:
        case "TSetGroupKickReq":
            return Frame_FrameType.TSetGroupKickReq;
        case 209:
        case "TSetGroupBanReq":
            return Frame_FrameType.TSetGroupBanReq;
        case 210:
        case "TSetGroupAnonymousReq":
            return Frame_FrameType.TSetGroupAnonymousReq;
        case 211:
        case "TSetGroupWholeBanReq":
            return Frame_FrameType.TSetGroupWholeBanReq;
        case 212:
        case "TSetGroupAdminReq":
            return Frame_FrameType.TSetGroupAdminReq;
        case 213:
        case "TSetGroupAnonymousBanReq":
            return Frame_FrameType.TSetGroupAnonymousBanReq;
        case 214:
        case "TSetGroupCardReq":
            return Frame_FrameType.TSetGroupCardReq;
        case 215:
        case "TSetGroupNameReq":
            return Frame_FrameType.TSetGroupNameReq;
        case 216:
        case "TSetGroupLeaveReq":
            return Frame_FrameType.TSetGroupLeaveReq;
        case 217:
        case "TSetGroupSpecialTitleReq":
            return Frame_FrameType.TSetGroupSpecialTitleReq;
        case 218:
        case "TSetFriendAddRequestReq":
            return Frame_FrameType.TSetFriendAddRequestReq;
        case 219:
        case "TSetGroupAddRequestReq":
            return Frame_FrameType.TSetGroupAddRequestReq;
        case 220:
        case "TGetLoginInfoReq":
            return Frame_FrameType.TGetLoginInfoReq;
        case 221:
        case "TGetStrangerInfoReq":
            return Frame_FrameType.TGetStrangerInfoReq;
        case 222:
        case "TGetFriendListReq":
            return Frame_FrameType.TGetFriendListReq;
        case 223:
        case "TGetGroupInfoReq":
            return Frame_FrameType.TGetGroupInfoReq;
        case 224:
        case "TGetGroupListReq":
            return Frame_FrameType.TGetGroupListReq;
        case 225:
        case "TGetGroupMemberInfoReq":
            return Frame_FrameType.TGetGroupMemberInfoReq;
        case 226:
        case "TGetGroupMemberListReq":
            return Frame_FrameType.TGetGroupMemberListReq;
        case 227:
        case "TGetGroupHonorInfoReq":
            return Frame_FrameType.TGetGroupHonorInfoReq;
        case 228:
        case "TGetCookiesReq":
            return Frame_FrameType.TGetCookiesReq;
        case 229:
        case "TGetCsrfTokenReq":
            return Frame_FrameType.TGetCsrfTokenReq;
        case 230:
        case "TGetCredentialsReq":
            return Frame_FrameType.TGetCredentialsReq;
        case 231:
        case "TGetRecordReq":
            return Frame_FrameType.TGetRecordReq;
        case 232:
        case "TGetImageReq":
            return Frame_FrameType.TGetImageReq;
        case 233:
        case "TCanSendImageReq":
            return Frame_FrameType.TCanSendImageReq;
        case 234:
        case "TCanSendRecordReq":
            return Frame_FrameType.TCanSendRecordReq;
        case 235:
        case "TGetStatusReq":
            return Frame_FrameType.TGetStatusReq;
        case 236:
        case "TGetVersionInfoReq":
            return Frame_FrameType.TGetVersionInfoReq;
        case 237:
        case "TSetRestartReq":
            return Frame_FrameType.TSetRestartReq;
        case 238:
        case "TCleanCacheReq":
            return Frame_FrameType.TCleanCacheReq;
        case 239:
        case "TSetGroupSignInReq":
            return Frame_FrameType.TSetGroupSignInReq;
        case 301:
        case "TSendPrivateMsgResp":
            return Frame_FrameType.TSendPrivateMsgResp;
        case 302:
        case "TSendGroupMsgResp":
            return Frame_FrameType.TSendGroupMsgResp;
        case 303:
        case "TSendMsgResp":
            return Frame_FrameType.TSendMsgResp;
        case 304:
        case "TDeleteMsgResp":
            return Frame_FrameType.TDeleteMsgResp;
        case 305:
        case "TGetMsgResp":
            return Frame_FrameType.TGetMsgResp;
        case 306:
        case "TGetForwardMsgResp":
            return Frame_FrameType.TGetForwardMsgResp;
        case 307:
        case "TSendLikeResp":
            return Frame_FrameType.TSendLikeResp;
        case 308:
        case "TSetGroupKickResp":
            return Frame_FrameType.TSetGroupKickResp;
        case 309:
        case "TSetGroupBanResp":
            return Frame_FrameType.TSetGroupBanResp;
        case 310:
        case "TSetGroupAnonymousResp":
            return Frame_FrameType.TSetGroupAnonymousResp;
        case 311:
        case "TSetGroupWholeBanResp":
            return Frame_FrameType.TSetGroupWholeBanResp;
        case 312:
        case "TSetGroupAdminResp":
            return Frame_FrameType.TSetGroupAdminResp;
        case 313:
        case "TSetGroupAnonymousBanResp":
            return Frame_FrameType.TSetGroupAnonymousBanResp;
        case 314:
        case "TSetGroupCardResp":
            return Frame_FrameType.TSetGroupCardResp;
        case 315:
        case "TSetGroupNameResp":
            return Frame_FrameType.TSetGroupNameResp;
        case 316:
        case "TSetGroupLeaveResp":
            return Frame_FrameType.TSetGroupLeaveResp;
        case 317:
        case "TSetGroupSpecialTitleResp":
            return Frame_FrameType.TSetGroupSpecialTitleResp;
        case 318:
        case "TSetFriendAddRequestResp":
            return Frame_FrameType.TSetFriendAddRequestResp;
        case 319:
        case "TSetGroupAddRequestResp":
            return Frame_FrameType.TSetGroupAddRequestResp;
        case 320:
        case "TGetLoginInfoResp":
            return Frame_FrameType.TGetLoginInfoResp;
        case 321:
        case "TGetStrangerInfoResp":
            return Frame_FrameType.TGetStrangerInfoResp;
        case 322:
        case "TGetFriendListResp":
            return Frame_FrameType.TGetFriendListResp;
        case 323:
        case "TGetGroupInfoResp":
            return Frame_FrameType.TGetGroupInfoResp;
        case 324:
        case "TGetGroupListResp":
            return Frame_FrameType.TGetGroupListResp;
        case 325:
        case "TGetGroupMemberInfoResp":
            return Frame_FrameType.TGetGroupMemberInfoResp;
        case 326:
        case "TGetGroupMemberListResp":
            return Frame_FrameType.TGetGroupMemberListResp;
        case 327:
        case "TGetGroupHonorInfoResp":
            return Frame_FrameType.TGetGroupHonorInfoResp;
        case 328:
        case "TGetCookiesResp":
            return Frame_FrameType.TGetCookiesResp;
        case 329:
        case "TGetCsrfTokenResp":
            return Frame_FrameType.TGetCsrfTokenResp;
        case 330:
        case "TGetCredentialsResp":
            return Frame_FrameType.TGetCredentialsResp;
        case 331:
        case "TGetRecordResp":
            return Frame_FrameType.TGetRecordResp;
        case 332:
        case "TGetImageResp":
            return Frame_FrameType.TGetImageResp;
        case 333:
        case "TCanSendImageResp":
            return Frame_FrameType.TCanSendImageResp;
        case 334:
        case "TCanSendRecordResp":
            return Frame_FrameType.TCanSendRecordResp;
        case 335:
        case "TGetStatusResp":
            return Frame_FrameType.TGetStatusResp;
        case 336:
        case "TGetVersionInfoResp":
            return Frame_FrameType.TGetVersionInfoResp;
        case 337:
        case "TSetRestartResp":
            return Frame_FrameType.TSetRestartResp;
        case 338:
        case "TCleanCacheResp":
            return Frame_FrameType.TCleanCacheResp;
        case 339:
        case "TSetGroupSignInResp":
            return Frame_FrameType.TSetGroupSignInResp;
        case -1:
        case "UNRECOGNIZED":
        default:
            return Frame_FrameType.UNRECOGNIZED;
    }
}
exports.frame_FrameTypeFromJSON = frame_FrameTypeFromJSON;
function frame_FrameTypeToJSON(object) {
    switch (object) {
        case Frame_FrameType.TUNKNOWN:
            return "TUNKNOWN";
        case Frame_FrameType.TPrivateMessageEvent:
            return "TPrivateMessageEvent";
        case Frame_FrameType.TGroupMessageEvent:
            return "TGroupMessageEvent";
        case Frame_FrameType.TGroupUploadNoticeEvent:
            return "TGroupUploadNoticeEvent";
        case Frame_FrameType.TGroupAdminNoticeEvent:
            return "TGroupAdminNoticeEvent";
        case Frame_FrameType.TGroupDecreaseNoticeEvent:
            return "TGroupDecreaseNoticeEvent";
        case Frame_FrameType.TGroupIncreaseNoticeEvent:
            return "TGroupIncreaseNoticeEvent";
        case Frame_FrameType.TGroupBanNoticeEvent:
            return "TGroupBanNoticeEvent";
        case Frame_FrameType.TFriendAddNoticeEvent:
            return "TFriendAddNoticeEvent";
        case Frame_FrameType.TGroupRecallNoticeEvent:
            return "TGroupRecallNoticeEvent";
        case Frame_FrameType.TFriendRecallNoticeEvent:
            return "TFriendRecallNoticeEvent";
        case Frame_FrameType.TFriendRequestEvent:
            return "TFriendRequestEvent";
        case Frame_FrameType.TGroupRequestEvent:
            return "TGroupRequestEvent";
        case Frame_FrameType.TSendPrivateMsgReq:
            return "TSendPrivateMsgReq";
        case Frame_FrameType.TSendGroupMsgReq:
            return "TSendGroupMsgReq";
        case Frame_FrameType.TSendMsgReq:
            return "TSendMsgReq";
        case Frame_FrameType.TDeleteMsgReq:
            return "TDeleteMsgReq";
        case Frame_FrameType.TGetMsgReq:
            return "TGetMsgReq";
        case Frame_FrameType.TGetForwardMsgReq:
            return "TGetForwardMsgReq";
        case Frame_FrameType.TSendLikeReq:
            return "TSendLikeReq";
        case Frame_FrameType.TSetGroupKickReq:
            return "TSetGroupKickReq";
        case Frame_FrameType.TSetGroupBanReq:
            return "TSetGroupBanReq";
        case Frame_FrameType.TSetGroupAnonymousReq:
            return "TSetGroupAnonymousReq";
        case Frame_FrameType.TSetGroupWholeBanReq:
            return "TSetGroupWholeBanReq";
        case Frame_FrameType.TSetGroupAdminReq:
            return "TSetGroupAdminReq";
        case Frame_FrameType.TSetGroupAnonymousBanReq:
            return "TSetGroupAnonymousBanReq";
        case Frame_FrameType.TSetGroupCardReq:
            return "TSetGroupCardReq";
        case Frame_FrameType.TSetGroupNameReq:
            return "TSetGroupNameReq";
        case Frame_FrameType.TSetGroupLeaveReq:
            return "TSetGroupLeaveReq";
        case Frame_FrameType.TSetGroupSpecialTitleReq:
            return "TSetGroupSpecialTitleReq";
        case Frame_FrameType.TSetFriendAddRequestReq:
            return "TSetFriendAddRequestReq";
        case Frame_FrameType.TSetGroupAddRequestReq:
            return "TSetGroupAddRequestReq";
        case Frame_FrameType.TGetLoginInfoReq:
            return "TGetLoginInfoReq";
        case Frame_FrameType.TGetStrangerInfoReq:
            return "TGetStrangerInfoReq";
        case Frame_FrameType.TGetFriendListReq:
            return "TGetFriendListReq";
        case Frame_FrameType.TGetGroupInfoReq:
            return "TGetGroupInfoReq";
        case Frame_FrameType.TGetGroupListReq:
            return "TGetGroupListReq";
        case Frame_FrameType.TGetGroupMemberInfoReq:
            return "TGetGroupMemberInfoReq";
        case Frame_FrameType.TGetGroupMemberListReq:
            return "TGetGroupMemberListReq";
        case Frame_FrameType.TGetGroupHonorInfoReq:
            return "TGetGroupHonorInfoReq";
        case Frame_FrameType.TGetCookiesReq:
            return "TGetCookiesReq";
        case Frame_FrameType.TGetCsrfTokenReq:
            return "TGetCsrfTokenReq";
        case Frame_FrameType.TGetCredentialsReq:
            return "TGetCredentialsReq";
        case Frame_FrameType.TGetRecordReq:
            return "TGetRecordReq";
        case Frame_FrameType.TGetImageReq:
            return "TGetImageReq";
        case Frame_FrameType.TCanSendImageReq:
            return "TCanSendImageReq";
        case Frame_FrameType.TCanSendRecordReq:
            return "TCanSendRecordReq";
        case Frame_FrameType.TGetStatusReq:
            return "TGetStatusReq";
        case Frame_FrameType.TGetVersionInfoReq:
            return "TGetVersionInfoReq";
        case Frame_FrameType.TSetRestartReq:
            return "TSetRestartReq";
        case Frame_FrameType.TCleanCacheReq:
            return "TCleanCacheReq";
        case Frame_FrameType.TSetGroupSignInReq:
            return "TSetGroupSignInReq";
        case Frame_FrameType.TSendPrivateMsgResp:
            return "TSendPrivateMsgResp";
        case Frame_FrameType.TSendGroupMsgResp:
            return "TSendGroupMsgResp";
        case Frame_FrameType.TSendMsgResp:
            return "TSendMsgResp";
        case Frame_FrameType.TDeleteMsgResp:
            return "TDeleteMsgResp";
        case Frame_FrameType.TGetMsgResp:
            return "TGetMsgResp";
        case Frame_FrameType.TGetForwardMsgResp:
            return "TGetForwardMsgResp";
        case Frame_FrameType.TSendLikeResp:
            return "TSendLikeResp";
        case Frame_FrameType.TSetGroupKickResp:
            return "TSetGroupKickResp";
        case Frame_FrameType.TSetGroupBanResp:
            return "TSetGroupBanResp";
        case Frame_FrameType.TSetGroupAnonymousResp:
            return "TSetGroupAnonymousResp";
        case Frame_FrameType.TSetGroupWholeBanResp:
            return "TSetGroupWholeBanResp";
        case Frame_FrameType.TSetGroupAdminResp:
            return "TSetGroupAdminResp";
        case Frame_FrameType.TSetGroupAnonymousBanResp:
            return "TSetGroupAnonymousBanResp";
        case Frame_FrameType.TSetGroupCardResp:
            return "TSetGroupCardResp";
        case Frame_FrameType.TSetGroupNameResp:
            return "TSetGroupNameResp";
        case Frame_FrameType.TSetGroupLeaveResp:
            return "TSetGroupLeaveResp";
        case Frame_FrameType.TSetGroupSpecialTitleResp:
            return "TSetGroupSpecialTitleResp";
        case Frame_FrameType.TSetFriendAddRequestResp:
            return "TSetFriendAddRequestResp";
        case Frame_FrameType.TSetGroupAddRequestResp:
            return "TSetGroupAddRequestResp";
        case Frame_FrameType.TGetLoginInfoResp:
            return "TGetLoginInfoResp";
        case Frame_FrameType.TGetStrangerInfoResp:
            return "TGetStrangerInfoResp";
        case Frame_FrameType.TGetFriendListResp:
            return "TGetFriendListResp";
        case Frame_FrameType.TGetGroupInfoResp:
            return "TGetGroupInfoResp";
        case Frame_FrameType.TGetGroupListResp:
            return "TGetGroupListResp";
        case Frame_FrameType.TGetGroupMemberInfoResp:
            return "TGetGroupMemberInfoResp";
        case Frame_FrameType.TGetGroupMemberListResp:
            return "TGetGroupMemberListResp";
        case Frame_FrameType.TGetGroupHonorInfoResp:
            return "TGetGroupHonorInfoResp";
        case Frame_FrameType.TGetCookiesResp:
            return "TGetCookiesResp";
        case Frame_FrameType.TGetCsrfTokenResp:
            return "TGetCsrfTokenResp";
        case Frame_FrameType.TGetCredentialsResp:
            return "TGetCredentialsResp";
        case Frame_FrameType.TGetRecordResp:
            return "TGetRecordResp";
        case Frame_FrameType.TGetImageResp:
            return "TGetImageResp";
        case Frame_FrameType.TCanSendImageResp:
            return "TCanSendImageResp";
        case Frame_FrameType.TCanSendRecordResp:
            return "TCanSendRecordResp";
        case Frame_FrameType.TGetStatusResp:
            return "TGetStatusResp";
        case Frame_FrameType.TGetVersionInfoResp:
            return "TGetVersionInfoResp";
        case Frame_FrameType.TSetRestartResp:
            return "TSetRestartResp";
        case Frame_FrameType.TCleanCacheResp:
            return "TCleanCacheResp";
        case Frame_FrameType.TSetGroupSignInResp:
            return "TSetGroupSignInResp";
        case Frame_FrameType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.frame_FrameTypeToJSON = frame_FrameTypeToJSON;
function createBaseFrame() {
    return {
        botId: 0,
        frameType: 0,
        echo: "",
        ok: false,
        extra: {},
        privateMessageEvent: undefined,
        groupMessageEvent: undefined,
        groupUploadNoticeEvent: undefined,
        groupAdminNoticeEvent: undefined,
        groupDecreaseNoticeEvent: undefined,
        groupIncreaseNoticeEvent: undefined,
        groupBanNoticeEvent: undefined,
        friendAddNoticeEvent: undefined,
        groupRecallNoticeEvent: undefined,
        friendRecallNoticeEvent: undefined,
        friendRequestEvent: undefined,
        groupRequestEvent: undefined,
        sendPrivateMsgReq: undefined,
        sendGroupMsgReq: undefined,
        sendMsgReq: undefined,
        deleteMsgReq: undefined,
        getMsgReq: undefined,
        getForwardMsgReq: undefined,
        sendLikeReq: undefined,
        setGroupKickReq: undefined,
        setGroupBanReq: undefined,
        setGroupAnonymousBanReq: undefined,
        setGroupWholeBanReq: undefined,
        setGroupAdminReq: undefined,
        setGroupAnonymousReq: undefined,
        setGroupCardReq: undefined,
        setGroupNameReq: undefined,
        setGroupLeaveReq: undefined,
        setGroupSpecialTitleReq: undefined,
        setFriendAddRequestReq: undefined,
        setGroupAddRequestReq: undefined,
        getLoginInfoReq: undefined,
        getStrangerInfoReq: undefined,
        getFriendListReq: undefined,
        getGroupInfoReq: undefined,
        getGroupListReq: undefined,
        getGroupMemberInfoReq: undefined,
        getGroupMemberListReq: undefined,
        getGroupHonorInfoReq: undefined,
        getCookiesReq: undefined,
        getCsrfTokenReq: undefined,
        getCredentialsReq: undefined,
        getRecordReq: undefined,
        getImageReq: undefined,
        canSendImageReq: undefined,
        canSendRecordReq: undefined,
        getStatusReq: undefined,
        getVersionInfoReq: undefined,
        setRestartReq: undefined,
        cleanCacheReq: undefined,
        setGroupSignInReq: undefined,
        sendPrivateMsgResp: undefined,
        sendGroupMsgResp: undefined,
        sendMsgResp: undefined,
        deleteMsgResp: undefined,
        getMsgResp: undefined,
        getForwardMsgResp: undefined,
        sendLikeResp: undefined,
        setGroupKickResp: undefined,
        setGroupBanResp: undefined,
        setGroupAnonymousBanResp: undefined,
        setGroupWholeBanResp: undefined,
        setGroupAdminResp: undefined,
        setGroupAnonymousResp: undefined,
        setGroupCardResp: undefined,
        setGroupNameResp: undefined,
        setGroupLeaveResp: undefined,
        setGroupSpecialTitleResp: undefined,
        setFriendAddRequestResp: undefined,
        setGroupAddRequestResp: undefined,
        getLoginInfoResp: undefined,
        getStrangerInfoResp: undefined,
        getFriendListResp: undefined,
        getGroupInfoResp: undefined,
        getGroupListResp: undefined,
        getGroupMemberInfoResp: undefined,
        getGroupMemberListResp: undefined,
        getGroupHonorInfoResp: undefined,
        getCookiesResp: undefined,
        getCsrfTokenResp: undefined,
        getCredentialsResp: undefined,
        getRecordResp: undefined,
        getImageResp: undefined,
        canSendImageResp: undefined,
        canSendRecordResp: undefined,
        getStatusResp: undefined,
        getVersionInfoResp: undefined,
        setRestartResp: undefined,
        cleanCacheResp: undefined,
        setGroupSignInResp: undefined,
    };
}
exports.Frame = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = _m0.Writer.create(); }
        if (message.botId !== 0) {
            writer.uint32(8).int64(message.botId);
        }
        if (message.frameType !== 0) {
            writer.uint32(16).int32(message.frameType);
        }
        if (message.echo !== "") {
            writer.uint32(26).string(message.echo);
        }
        if (message.ok === true) {
            writer.uint32(32).bool(message.ok);
        }
        Object.entries(message.extra).forEach(function (_a) {
            var key = _a[0], value = _a[1];
            exports.Frame_ExtraEntry.encode({ key: key, value: value }, writer.uint32(802).fork()).ldelim();
        });
        if (message.privateMessageEvent !== undefined) {
            onebot_event_1.PrivateMessageEvent.encode(message.privateMessageEvent, writer.uint32(810).fork()).ldelim();
        }
        if (message.groupMessageEvent !== undefined) {
            onebot_event_1.GroupMessageEvent.encode(message.groupMessageEvent, writer.uint32(818).fork()).ldelim();
        }
        if (message.groupUploadNoticeEvent !== undefined) {
            onebot_event_1.GroupUploadNoticeEvent.encode(message.groupUploadNoticeEvent, writer.uint32(826).fork()).ldelim();
        }
        if (message.groupAdminNoticeEvent !== undefined) {
            onebot_event_1.GroupAdminNoticeEvent.encode(message.groupAdminNoticeEvent, writer.uint32(834).fork()).ldelim();
        }
        if (message.groupDecreaseNoticeEvent !== undefined) {
            onebot_event_1.GroupDecreaseNoticeEvent.encode(message.groupDecreaseNoticeEvent, writer.uint32(842).fork()).ldelim();
        }
        if (message.groupIncreaseNoticeEvent !== undefined) {
            onebot_event_1.GroupIncreaseNoticeEvent.encode(message.groupIncreaseNoticeEvent, writer.uint32(850).fork()).ldelim();
        }
        if (message.groupBanNoticeEvent !== undefined) {
            onebot_event_1.GroupBanNoticeEvent.encode(message.groupBanNoticeEvent, writer.uint32(858).fork()).ldelim();
        }
        if (message.friendAddNoticeEvent !== undefined) {
            onebot_event_1.FriendAddNoticeEvent.encode(message.friendAddNoticeEvent, writer.uint32(866).fork()).ldelim();
        }
        if (message.groupRecallNoticeEvent !== undefined) {
            onebot_event_1.GroupRecallNoticeEvent.encode(message.groupRecallNoticeEvent, writer.uint32(874).fork()).ldelim();
        }
        if (message.friendRecallNoticeEvent !== undefined) {
            onebot_event_1.FriendRecallNoticeEvent.encode(message.friendRecallNoticeEvent, writer.uint32(882).fork()).ldelim();
        }
        if (message.friendRequestEvent !== undefined) {
            onebot_event_1.FriendRequestEvent.encode(message.friendRequestEvent, writer.uint32(890).fork()).ldelim();
        }
        if (message.groupRequestEvent !== undefined) {
            onebot_event_1.GroupRequestEvent.encode(message.groupRequestEvent, writer.uint32(898).fork()).ldelim();
        }
        if (message.sendPrivateMsgReq !== undefined) {
            onebot_api_1.SendPrivateMsgReq.encode(message.sendPrivateMsgReq, writer.uint32(1610).fork()).ldelim();
        }
        if (message.sendGroupMsgReq !== undefined) {
            onebot_api_1.SendGroupMsgReq.encode(message.sendGroupMsgReq, writer.uint32(1618).fork()).ldelim();
        }
        if (message.sendMsgReq !== undefined) {
            onebot_api_1.SendMsgReq.encode(message.sendMsgReq, writer.uint32(1626).fork()).ldelim();
        }
        if (message.deleteMsgReq !== undefined) {
            onebot_api_1.DeleteMsgReq.encode(message.deleteMsgReq, writer.uint32(1634).fork()).ldelim();
        }
        if (message.getMsgReq !== undefined) {
            onebot_api_1.GetMsgReq.encode(message.getMsgReq, writer.uint32(1642).fork()).ldelim();
        }
        if (message.getForwardMsgReq !== undefined) {
            onebot_api_1.GetForwardMsgReq.encode(message.getForwardMsgReq, writer.uint32(1650).fork()).ldelim();
        }
        if (message.sendLikeReq !== undefined) {
            onebot_api_1.SendLikeReq.encode(message.sendLikeReq, writer.uint32(1658).fork()).ldelim();
        }
        if (message.setGroupKickReq !== undefined) {
            onebot_api_1.SetGroupKickReq.encode(message.setGroupKickReq, writer.uint32(1666).fork()).ldelim();
        }
        if (message.setGroupBanReq !== undefined) {
            onebot_api_1.SetGroupBanReq.encode(message.setGroupBanReq, writer.uint32(1674).fork()).ldelim();
        }
        if (message.setGroupAnonymousBanReq !== undefined) {
            onebot_api_1.SetGroupAnonymousBanReq.encode(message.setGroupAnonymousBanReq, writer.uint32(1682).fork()).ldelim();
        }
        if (message.setGroupWholeBanReq !== undefined) {
            onebot_api_1.SetGroupWholeBanReq.encode(message.setGroupWholeBanReq, writer.uint32(1690).fork()).ldelim();
        }
        if (message.setGroupAdminReq !== undefined) {
            onebot_api_1.SetGroupAdminReq.encode(message.setGroupAdminReq, writer.uint32(1698).fork()).ldelim();
        }
        if (message.setGroupAnonymousReq !== undefined) {
            onebot_api_1.SetGroupAnonymousReq.encode(message.setGroupAnonymousReq, writer.uint32(1706).fork()).ldelim();
        }
        if (message.setGroupCardReq !== undefined) {
            onebot_api_1.SetGroupCardReq.encode(message.setGroupCardReq, writer.uint32(1714).fork()).ldelim();
        }
        if (message.setGroupNameReq !== undefined) {
            onebot_api_1.SetGroupNameReq.encode(message.setGroupNameReq, writer.uint32(1722).fork()).ldelim();
        }
        if (message.setGroupLeaveReq !== undefined) {
            onebot_api_1.SetGroupLeaveReq.encode(message.setGroupLeaveReq, writer.uint32(1730).fork()).ldelim();
        }
        if (message.setGroupSpecialTitleReq !== undefined) {
            onebot_api_1.SetGroupSpecialTitleReq.encode(message.setGroupSpecialTitleReq, writer.uint32(1738).fork()).ldelim();
        }
        if (message.setFriendAddRequestReq !== undefined) {
            onebot_api_1.SetFriendAddRequestReq.encode(message.setFriendAddRequestReq, writer.uint32(1746).fork()).ldelim();
        }
        if (message.setGroupAddRequestReq !== undefined) {
            onebot_api_1.SetGroupAddRequestReq.encode(message.setGroupAddRequestReq, writer.uint32(1754).fork()).ldelim();
        }
        if (message.getLoginInfoReq !== undefined) {
            onebot_api_1.GetLoginInfoReq.encode(message.getLoginInfoReq, writer.uint32(1762).fork()).ldelim();
        }
        if (message.getStrangerInfoReq !== undefined) {
            onebot_api_1.GetStrangerInfoReq.encode(message.getStrangerInfoReq, writer.uint32(1770).fork()).ldelim();
        }
        if (message.getFriendListReq !== undefined) {
            onebot_api_1.GetFriendListReq.encode(message.getFriendListReq, writer.uint32(1778).fork()).ldelim();
        }
        if (message.getGroupInfoReq !== undefined) {
            onebot_api_1.GetGroupInfoReq.encode(message.getGroupInfoReq, writer.uint32(1786).fork()).ldelim();
        }
        if (message.getGroupListReq !== undefined) {
            onebot_api_1.GetGroupListReq.encode(message.getGroupListReq, writer.uint32(1794).fork()).ldelim();
        }
        if (message.getGroupMemberInfoReq !== undefined) {
            onebot_api_1.GetGroupMemberInfoReq.encode(message.getGroupMemberInfoReq, writer.uint32(1802).fork()).ldelim();
        }
        if (message.getGroupMemberListReq !== undefined) {
            onebot_api_1.GetGroupMemberListReq.encode(message.getGroupMemberListReq, writer.uint32(1810).fork()).ldelim();
        }
        if (message.getGroupHonorInfoReq !== undefined) {
            onebot_api_1.GetGroupHonorInfoReq.encode(message.getGroupHonorInfoReq, writer.uint32(1818).fork()).ldelim();
        }
        if (message.getCookiesReq !== undefined) {
            onebot_api_1.GetCookiesReq.encode(message.getCookiesReq, writer.uint32(1826).fork()).ldelim();
        }
        if (message.getCsrfTokenReq !== undefined) {
            onebot_api_1.GetCsrfTokenReq.encode(message.getCsrfTokenReq, writer.uint32(1834).fork()).ldelim();
        }
        if (message.getCredentialsReq !== undefined) {
            onebot_api_1.GetCredentialsReq.encode(message.getCredentialsReq, writer.uint32(1842).fork()).ldelim();
        }
        if (message.getRecordReq !== undefined) {
            onebot_api_1.GetRecordReq.encode(message.getRecordReq, writer.uint32(1850).fork()).ldelim();
        }
        if (message.getImageReq !== undefined) {
            onebot_api_1.GetImageReq.encode(message.getImageReq, writer.uint32(1858).fork()).ldelim();
        }
        if (message.canSendImageReq !== undefined) {
            onebot_api_1.CanSendImageReq.encode(message.canSendImageReq, writer.uint32(1866).fork()).ldelim();
        }
        if (message.canSendRecordReq !== undefined) {
            onebot_api_1.CanSendRecordReq.encode(message.canSendRecordReq, writer.uint32(1874).fork()).ldelim();
        }
        if (message.getStatusReq !== undefined) {
            onebot_api_1.GetStatusReq.encode(message.getStatusReq, writer.uint32(1882).fork()).ldelim();
        }
        if (message.getVersionInfoReq !== undefined) {
            onebot_api_1.GetVersionInfoReq.encode(message.getVersionInfoReq, writer.uint32(1890).fork()).ldelim();
        }
        if (message.setRestartReq !== undefined) {
            onebot_api_1.SetRestartReq.encode(message.setRestartReq, writer.uint32(1898).fork()).ldelim();
        }
        if (message.cleanCacheReq !== undefined) {
            onebot_api_1.CleanCacheReq.encode(message.cleanCacheReq, writer.uint32(1906).fork()).ldelim();
        }
        if (message.setGroupSignInReq !== undefined) {
            onebot_api_1.SetGroupSignInReq.encode(message.setGroupSignInReq, writer.uint32(1914).fork()).ldelim();
        }
        if (message.sendPrivateMsgResp !== undefined) {
            onebot_api_1.SendPrivateMsgResp.encode(message.sendPrivateMsgResp, writer.uint32(2410).fork()).ldelim();
        }
        if (message.sendGroupMsgResp !== undefined) {
            onebot_api_1.SendGroupMsgResp.encode(message.sendGroupMsgResp, writer.uint32(2418).fork()).ldelim();
        }
        if (message.sendMsgResp !== undefined) {
            onebot_api_1.SendMsgResp.encode(message.sendMsgResp, writer.uint32(2426).fork()).ldelim();
        }
        if (message.deleteMsgResp !== undefined) {
            onebot_api_1.DeleteMsgResp.encode(message.deleteMsgResp, writer.uint32(2434).fork()).ldelim();
        }
        if (message.getMsgResp !== undefined) {
            onebot_api_1.GetMsgResp.encode(message.getMsgResp, writer.uint32(2442).fork()).ldelim();
        }
        if (message.getForwardMsgResp !== undefined) {
            onebot_api_1.GetForwardMsgResp.encode(message.getForwardMsgResp, writer.uint32(2450).fork()).ldelim();
        }
        if (message.sendLikeResp !== undefined) {
            onebot_api_1.SendLikeResp.encode(message.sendLikeResp, writer.uint32(2458).fork()).ldelim();
        }
        if (message.setGroupKickResp !== undefined) {
            onebot_api_1.SetGroupKickResp.encode(message.setGroupKickResp, writer.uint32(2466).fork()).ldelim();
        }
        if (message.setGroupBanResp !== undefined) {
            onebot_api_1.SetGroupBanResp.encode(message.setGroupBanResp, writer.uint32(2474).fork()).ldelim();
        }
        if (message.setGroupAnonymousBanResp !== undefined) {
            onebot_api_1.SetGroupAnonymousBanResp.encode(message.setGroupAnonymousBanResp, writer.uint32(2482).fork()).ldelim();
        }
        if (message.setGroupWholeBanResp !== undefined) {
            onebot_api_1.SetGroupWholeBanResp.encode(message.setGroupWholeBanResp, writer.uint32(2490).fork()).ldelim();
        }
        if (message.setGroupAdminResp !== undefined) {
            onebot_api_1.SetGroupAdminResp.encode(message.setGroupAdminResp, writer.uint32(2498).fork()).ldelim();
        }
        if (message.setGroupAnonymousResp !== undefined) {
            onebot_api_1.SetGroupAnonymousResp.encode(message.setGroupAnonymousResp, writer.uint32(2506).fork()).ldelim();
        }
        if (message.setGroupCardResp !== undefined) {
            onebot_api_1.SetGroupCardResp.encode(message.setGroupCardResp, writer.uint32(2514).fork()).ldelim();
        }
        if (message.setGroupNameResp !== undefined) {
            onebot_api_1.SetGroupNameResp.encode(message.setGroupNameResp, writer.uint32(2522).fork()).ldelim();
        }
        if (message.setGroupLeaveResp !== undefined) {
            onebot_api_1.SetGroupLeaveResp.encode(message.setGroupLeaveResp, writer.uint32(2530).fork()).ldelim();
        }
        if (message.setGroupSpecialTitleResp !== undefined) {
            onebot_api_1.SetGroupSpecialTitleResp.encode(message.setGroupSpecialTitleResp, writer.uint32(2538).fork()).ldelim();
        }
        if (message.setFriendAddRequestResp !== undefined) {
            onebot_api_1.SetFriendAddRequestResp.encode(message.setFriendAddRequestResp, writer.uint32(2546).fork()).ldelim();
        }
        if (message.setGroupAddRequestResp !== undefined) {
            onebot_api_1.SetGroupAddRequestResp.encode(message.setGroupAddRequestResp, writer.uint32(2554).fork()).ldelim();
        }
        if (message.getLoginInfoResp !== undefined) {
            onebot_api_1.GetLoginInfoResp.encode(message.getLoginInfoResp, writer.uint32(2562).fork()).ldelim();
        }
        if (message.getStrangerInfoResp !== undefined) {
            onebot_api_1.GetStrangerInfoResp.encode(message.getStrangerInfoResp, writer.uint32(2570).fork()).ldelim();
        }
        if (message.getFriendListResp !== undefined) {
            onebot_api_1.GetFriendListResp.encode(message.getFriendListResp, writer.uint32(2578).fork()).ldelim();
        }
        if (message.getGroupInfoResp !== undefined) {
            onebot_api_1.GetGroupInfoResp.encode(message.getGroupInfoResp, writer.uint32(2586).fork()).ldelim();
        }
        if (message.getGroupListResp !== undefined) {
            onebot_api_1.GetGroupListResp.encode(message.getGroupListResp, writer.uint32(2594).fork()).ldelim();
        }
        if (message.getGroupMemberInfoResp !== undefined) {
            onebot_api_1.GetGroupMemberInfoResp.encode(message.getGroupMemberInfoResp, writer.uint32(2602).fork()).ldelim();
        }
        if (message.getGroupMemberListResp !== undefined) {
            onebot_api_1.GetGroupMemberListResp.encode(message.getGroupMemberListResp, writer.uint32(2610).fork()).ldelim();
        }
        if (message.getGroupHonorInfoResp !== undefined) {
            onebot_api_1.GetGroupHonorInfoResp.encode(message.getGroupHonorInfoResp, writer.uint32(2618).fork()).ldelim();
        }
        if (message.getCookiesResp !== undefined) {
            onebot_api_1.GetCookiesResp.encode(message.getCookiesResp, writer.uint32(2626).fork()).ldelim();
        }
        if (message.getCsrfTokenResp !== undefined) {
            onebot_api_1.GetCsrfTokenResp.encode(message.getCsrfTokenResp, writer.uint32(2634).fork()).ldelim();
        }
        if (message.getCredentialsResp !== undefined) {
            onebot_api_1.GetCredentialsResp.encode(message.getCredentialsResp, writer.uint32(2642).fork()).ldelim();
        }
        if (message.getRecordResp !== undefined) {
            onebot_api_1.GetRecordResp.encode(message.getRecordResp, writer.uint32(2650).fork()).ldelim();
        }
        if (message.getImageResp !== undefined) {
            onebot_api_1.GetImageResp.encode(message.getImageResp, writer.uint32(2658).fork()).ldelim();
        }
        if (message.canSendImageResp !== undefined) {
            onebot_api_1.CanSendImageResp.encode(message.canSendImageResp, writer.uint32(2666).fork()).ldelim();
        }
        if (message.canSendRecordResp !== undefined) {
            onebot_api_1.CanSendRecordResp.encode(message.canSendRecordResp, writer.uint32(2674).fork()).ldelim();
        }
        if (message.getStatusResp !== undefined) {
            onebot_api_1.GetStatusResp.encode(message.getStatusResp, writer.uint32(2682).fork()).ldelim();
        }
        if (message.getVersionInfoResp !== undefined) {
            onebot_api_1.GetVersionInfoResp.encode(message.getVersionInfoResp, writer.uint32(2690).fork()).ldelim();
        }
        if (message.setRestartResp !== undefined) {
            onebot_api_1.SetRestartResp.encode(message.setRestartResp, writer.uint32(2698).fork()).ldelim();
        }
        if (message.cleanCacheResp !== undefined) {
            onebot_api_1.CleanCacheResp.encode(message.cleanCacheResp, writer.uint32(2706).fork()).ldelim();
        }
        if (message.setGroupSignInResp !== undefined) {
            onebot_api_1.SetGroupSignInResp.encode(message.setGroupSignInResp, writer.uint32(2714).fork()).ldelim();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseFrame();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.botId = longToNumber(reader.int64());
                    break;
                case 2:
                    message.frameType = reader.int32();
                    break;
                case 3:
                    message.echo = reader.string();
                    break;
                case 4:
                    message.ok = reader.bool();
                    break;
                case 100:
                    var entry100 = exports.Frame_ExtraEntry.decode(reader, reader.uint32());
                    if (entry100.value !== undefined) {
                        message.extra[entry100.key] = entry100.value;
                    }
                    break;
                case 101:
                    message.privateMessageEvent = onebot_event_1.PrivateMessageEvent.decode(reader, reader.uint32());
                    break;
                case 102:
                    message.groupMessageEvent = onebot_event_1.GroupMessageEvent.decode(reader, reader.uint32());
                    break;
                case 103:
                    message.groupUploadNoticeEvent = onebot_event_1.GroupUploadNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 104:
                    message.groupAdminNoticeEvent = onebot_event_1.GroupAdminNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 105:
                    message.groupDecreaseNoticeEvent = onebot_event_1.GroupDecreaseNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 106:
                    message.groupIncreaseNoticeEvent = onebot_event_1.GroupIncreaseNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 107:
                    message.groupBanNoticeEvent = onebot_event_1.GroupBanNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 108:
                    message.friendAddNoticeEvent = onebot_event_1.FriendAddNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 109:
                    message.groupRecallNoticeEvent = onebot_event_1.GroupRecallNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 110:
                    message.friendRecallNoticeEvent = onebot_event_1.FriendRecallNoticeEvent.decode(reader, reader.uint32());
                    break;
                case 111:
                    message.friendRequestEvent = onebot_event_1.FriendRequestEvent.decode(reader, reader.uint32());
                    break;
                case 112:
                    message.groupRequestEvent = onebot_event_1.GroupRequestEvent.decode(reader, reader.uint32());
                    break;
                case 201:
                    message.sendPrivateMsgReq = onebot_api_1.SendPrivateMsgReq.decode(reader, reader.uint32());
                    break;
                case 202:
                    message.sendGroupMsgReq = onebot_api_1.SendGroupMsgReq.decode(reader, reader.uint32());
                    break;
                case 203:
                    message.sendMsgReq = onebot_api_1.SendMsgReq.decode(reader, reader.uint32());
                    break;
                case 204:
                    message.deleteMsgReq = onebot_api_1.DeleteMsgReq.decode(reader, reader.uint32());
                    break;
                case 205:
                    message.getMsgReq = onebot_api_1.GetMsgReq.decode(reader, reader.uint32());
                    break;
                case 206:
                    message.getForwardMsgReq = onebot_api_1.GetForwardMsgReq.decode(reader, reader.uint32());
                    break;
                case 207:
                    message.sendLikeReq = onebot_api_1.SendLikeReq.decode(reader, reader.uint32());
                    break;
                case 208:
                    message.setGroupKickReq = onebot_api_1.SetGroupKickReq.decode(reader, reader.uint32());
                    break;
                case 209:
                    message.setGroupBanReq = onebot_api_1.SetGroupBanReq.decode(reader, reader.uint32());
                    break;
                case 210:
                    message.setGroupAnonymousBanReq = onebot_api_1.SetGroupAnonymousBanReq.decode(reader, reader.uint32());
                    break;
                case 211:
                    message.setGroupWholeBanReq = onebot_api_1.SetGroupWholeBanReq.decode(reader, reader.uint32());
                    break;
                case 212:
                    message.setGroupAdminReq = onebot_api_1.SetGroupAdminReq.decode(reader, reader.uint32());
                    break;
                case 213:
                    message.setGroupAnonymousReq = onebot_api_1.SetGroupAnonymousReq.decode(reader, reader.uint32());
                    break;
                case 214:
                    message.setGroupCardReq = onebot_api_1.SetGroupCardReq.decode(reader, reader.uint32());
                    break;
                case 215:
                    message.setGroupNameReq = onebot_api_1.SetGroupNameReq.decode(reader, reader.uint32());
                    break;
                case 216:
                    message.setGroupLeaveReq = onebot_api_1.SetGroupLeaveReq.decode(reader, reader.uint32());
                    break;
                case 217:
                    message.setGroupSpecialTitleReq = onebot_api_1.SetGroupSpecialTitleReq.decode(reader, reader.uint32());
                    break;
                case 218:
                    message.setFriendAddRequestReq = onebot_api_1.SetFriendAddRequestReq.decode(reader, reader.uint32());
                    break;
                case 219:
                    message.setGroupAddRequestReq = onebot_api_1.SetGroupAddRequestReq.decode(reader, reader.uint32());
                    break;
                case 220:
                    message.getLoginInfoReq = onebot_api_1.GetLoginInfoReq.decode(reader, reader.uint32());
                    break;
                case 221:
                    message.getStrangerInfoReq = onebot_api_1.GetStrangerInfoReq.decode(reader, reader.uint32());
                    break;
                case 222:
                    message.getFriendListReq = onebot_api_1.GetFriendListReq.decode(reader, reader.uint32());
                    break;
                case 223:
                    message.getGroupInfoReq = onebot_api_1.GetGroupInfoReq.decode(reader, reader.uint32());
                    break;
                case 224:
                    message.getGroupListReq = onebot_api_1.GetGroupListReq.decode(reader, reader.uint32());
                    break;
                case 225:
                    message.getGroupMemberInfoReq = onebot_api_1.GetGroupMemberInfoReq.decode(reader, reader.uint32());
                    break;
                case 226:
                    message.getGroupMemberListReq = onebot_api_1.GetGroupMemberListReq.decode(reader, reader.uint32());
                    break;
                case 227:
                    message.getGroupHonorInfoReq = onebot_api_1.GetGroupHonorInfoReq.decode(reader, reader.uint32());
                    break;
                case 228:
                    message.getCookiesReq = onebot_api_1.GetCookiesReq.decode(reader, reader.uint32());
                    break;
                case 229:
                    message.getCsrfTokenReq = onebot_api_1.GetCsrfTokenReq.decode(reader, reader.uint32());
                    break;
                case 230:
                    message.getCredentialsReq = onebot_api_1.GetCredentialsReq.decode(reader, reader.uint32());
                    break;
                case 231:
                    message.getRecordReq = onebot_api_1.GetRecordReq.decode(reader, reader.uint32());
                    break;
                case 232:
                    message.getImageReq = onebot_api_1.GetImageReq.decode(reader, reader.uint32());
                    break;
                case 233:
                    message.canSendImageReq = onebot_api_1.CanSendImageReq.decode(reader, reader.uint32());
                    break;
                case 234:
                    message.canSendRecordReq = onebot_api_1.CanSendRecordReq.decode(reader, reader.uint32());
                    break;
                case 235:
                    message.getStatusReq = onebot_api_1.GetStatusReq.decode(reader, reader.uint32());
                    break;
                case 236:
                    message.getVersionInfoReq = onebot_api_1.GetVersionInfoReq.decode(reader, reader.uint32());
                    break;
                case 237:
                    message.setRestartReq = onebot_api_1.SetRestartReq.decode(reader, reader.uint32());
                    break;
                case 238:
                    message.cleanCacheReq = onebot_api_1.CleanCacheReq.decode(reader, reader.uint32());
                    break;
                case 239:
                    message.setGroupSignInReq = onebot_api_1.SetGroupSignInReq.decode(reader, reader.uint32());
                    break;
                case 301:
                    message.sendPrivateMsgResp = onebot_api_1.SendPrivateMsgResp.decode(reader, reader.uint32());
                    break;
                case 302:
                    message.sendGroupMsgResp = onebot_api_1.SendGroupMsgResp.decode(reader, reader.uint32());
                    break;
                case 303:
                    message.sendMsgResp = onebot_api_1.SendMsgResp.decode(reader, reader.uint32());
                    break;
                case 304:
                    message.deleteMsgResp = onebot_api_1.DeleteMsgResp.decode(reader, reader.uint32());
                    break;
                case 305:
                    message.getMsgResp = onebot_api_1.GetMsgResp.decode(reader, reader.uint32());
                    break;
                case 306:
                    message.getForwardMsgResp = onebot_api_1.GetForwardMsgResp.decode(reader, reader.uint32());
                    break;
                case 307:
                    message.sendLikeResp = onebot_api_1.SendLikeResp.decode(reader, reader.uint32());
                    break;
                case 308:
                    message.setGroupKickResp = onebot_api_1.SetGroupKickResp.decode(reader, reader.uint32());
                    break;
                case 309:
                    message.setGroupBanResp = onebot_api_1.SetGroupBanResp.decode(reader, reader.uint32());
                    break;
                case 310:
                    message.setGroupAnonymousBanResp = onebot_api_1.SetGroupAnonymousBanResp.decode(reader, reader.uint32());
                    break;
                case 311:
                    message.setGroupWholeBanResp = onebot_api_1.SetGroupWholeBanResp.decode(reader, reader.uint32());
                    break;
                case 312:
                    message.setGroupAdminResp = onebot_api_1.SetGroupAdminResp.decode(reader, reader.uint32());
                    break;
                case 313:
                    message.setGroupAnonymousResp = onebot_api_1.SetGroupAnonymousResp.decode(reader, reader.uint32());
                    break;
                case 314:
                    message.setGroupCardResp = onebot_api_1.SetGroupCardResp.decode(reader, reader.uint32());
                    break;
                case 315:
                    message.setGroupNameResp = onebot_api_1.SetGroupNameResp.decode(reader, reader.uint32());
                    break;
                case 316:
                    message.setGroupLeaveResp = onebot_api_1.SetGroupLeaveResp.decode(reader, reader.uint32());
                    break;
                case 317:
                    message.setGroupSpecialTitleResp = onebot_api_1.SetGroupSpecialTitleResp.decode(reader, reader.uint32());
                    break;
                case 318:
                    message.setFriendAddRequestResp = onebot_api_1.SetFriendAddRequestResp.decode(reader, reader.uint32());
                    break;
                case 319:
                    message.setGroupAddRequestResp = onebot_api_1.SetGroupAddRequestResp.decode(reader, reader.uint32());
                    break;
                case 320:
                    message.getLoginInfoResp = onebot_api_1.GetLoginInfoResp.decode(reader, reader.uint32());
                    break;
                case 321:
                    message.getStrangerInfoResp = onebot_api_1.GetStrangerInfoResp.decode(reader, reader.uint32());
                    break;
                case 322:
                    message.getFriendListResp = onebot_api_1.GetFriendListResp.decode(reader, reader.uint32());
                    break;
                case 323:
                    message.getGroupInfoResp = onebot_api_1.GetGroupInfoResp.decode(reader, reader.uint32());
                    break;
                case 324:
                    message.getGroupListResp = onebot_api_1.GetGroupListResp.decode(reader, reader.uint32());
                    break;
                case 325:
                    message.getGroupMemberInfoResp = onebot_api_1.GetGroupMemberInfoResp.decode(reader, reader.uint32());
                    break;
                case 326:
                    message.getGroupMemberListResp = onebot_api_1.GetGroupMemberListResp.decode(reader, reader.uint32());
                    break;
                case 327:
                    message.getGroupHonorInfoResp = onebot_api_1.GetGroupHonorInfoResp.decode(reader, reader.uint32());
                    break;
                case 328:
                    message.getCookiesResp = onebot_api_1.GetCookiesResp.decode(reader, reader.uint32());
                    break;
                case 329:
                    message.getCsrfTokenResp = onebot_api_1.GetCsrfTokenResp.decode(reader, reader.uint32());
                    break;
                case 330:
                    message.getCredentialsResp = onebot_api_1.GetCredentialsResp.decode(reader, reader.uint32());
                    break;
                case 331:
                    message.getRecordResp = onebot_api_1.GetRecordResp.decode(reader, reader.uint32());
                    break;
                case 332:
                    message.getImageResp = onebot_api_1.GetImageResp.decode(reader, reader.uint32());
                    break;
                case 333:
                    message.canSendImageResp = onebot_api_1.CanSendImageResp.decode(reader, reader.uint32());
                    break;
                case 334:
                    message.canSendRecordResp = onebot_api_1.CanSendRecordResp.decode(reader, reader.uint32());
                    break;
                case 335:
                    message.getStatusResp = onebot_api_1.GetStatusResp.decode(reader, reader.uint32());
                    break;
                case 336:
                    message.getVersionInfoResp = onebot_api_1.GetVersionInfoResp.decode(reader, reader.uint32());
                    break;
                case 337:
                    message.setRestartResp = onebot_api_1.SetRestartResp.decode(reader, reader.uint32());
                    break;
                case 338:
                    message.cleanCacheResp = onebot_api_1.CleanCacheResp.decode(reader, reader.uint32());
                    break;
                case 339:
                    message.setGroupSignInResp = onebot_api_1.SetGroupSignInResp.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            botId: isSet(object.botId) ? Number(object.botId) : 0,
            frameType: isSet(object.frameType)
                ? frame_FrameTypeFromJSON(object.frameType)
                : 0,
            echo: isSet(object.echo) ? String(object.echo) : "",
            ok: isSet(object.ok) ? Boolean(object.ok) : false,
            extra: isObject(object.extra)
                ? Object.entries(object.extra).reduce(function (acc, _a) {
                    var key = _a[0], value = _a[1];
                    acc[key] = String(value);
                    return acc;
                }, {})
                : {},
            privateMessageEvent: isSet(object.privateMessageEvent)
                ? onebot_event_1.PrivateMessageEvent.fromJSON(object.privateMessageEvent)
                : undefined,
            groupMessageEvent: isSet(object.groupMessageEvent)
                ? onebot_event_1.GroupMessageEvent.fromJSON(object.groupMessageEvent)
                : undefined,
            groupUploadNoticeEvent: isSet(object.groupUploadNoticeEvent)
                ? onebot_event_1.GroupUploadNoticeEvent.fromJSON(object.groupUploadNoticeEvent)
                : undefined,
            groupAdminNoticeEvent: isSet(object.groupAdminNoticeEvent)
                ? onebot_event_1.GroupAdminNoticeEvent.fromJSON(object.groupAdminNoticeEvent)
                : undefined,
            groupDecreaseNoticeEvent: isSet(object.groupDecreaseNoticeEvent)
                ? onebot_event_1.GroupDecreaseNoticeEvent.fromJSON(object.groupDecreaseNoticeEvent)
                : undefined,
            groupIncreaseNoticeEvent: isSet(object.groupIncreaseNoticeEvent)
                ? onebot_event_1.GroupIncreaseNoticeEvent.fromJSON(object.groupIncreaseNoticeEvent)
                : undefined,
            groupBanNoticeEvent: isSet(object.groupBanNoticeEvent)
                ? onebot_event_1.GroupBanNoticeEvent.fromJSON(object.groupBanNoticeEvent)
                : undefined,
            friendAddNoticeEvent: isSet(object.friendAddNoticeEvent)
                ? onebot_event_1.FriendAddNoticeEvent.fromJSON(object.friendAddNoticeEvent)
                : undefined,
            groupRecallNoticeEvent: isSet(object.groupRecallNoticeEvent)
                ? onebot_event_1.GroupRecallNoticeEvent.fromJSON(object.groupRecallNoticeEvent)
                : undefined,
            friendRecallNoticeEvent: isSet(object.friendRecallNoticeEvent)
                ? onebot_event_1.FriendRecallNoticeEvent.fromJSON(object.friendRecallNoticeEvent)
                : undefined,
            friendRequestEvent: isSet(object.friendRequestEvent)
                ? onebot_event_1.FriendRequestEvent.fromJSON(object.friendRequestEvent)
                : undefined,
            groupRequestEvent: isSet(object.groupRequestEvent)
                ? onebot_event_1.GroupRequestEvent.fromJSON(object.groupRequestEvent)
                : undefined,
            sendPrivateMsgReq: isSet(object.sendPrivateMsgReq)
                ? onebot_api_1.SendPrivateMsgReq.fromJSON(object.sendPrivateMsgReq)
                : undefined,
            sendGroupMsgReq: isSet(object.sendGroupMsgReq)
                ? onebot_api_1.SendGroupMsgReq.fromJSON(object.sendGroupMsgReq)
                : undefined,
            sendMsgReq: isSet(object.sendMsgReq)
                ? onebot_api_1.SendMsgReq.fromJSON(object.sendMsgReq)
                : undefined,
            deleteMsgReq: isSet(object.deleteMsgReq)
                ? onebot_api_1.DeleteMsgReq.fromJSON(object.deleteMsgReq)
                : undefined,
            getMsgReq: isSet(object.getMsgReq)
                ? onebot_api_1.GetMsgReq.fromJSON(object.getMsgReq)
                : undefined,
            getForwardMsgReq: isSet(object.getForwardMsgReq)
                ? onebot_api_1.GetForwardMsgReq.fromJSON(object.getForwardMsgReq)
                : undefined,
            sendLikeReq: isSet(object.sendLikeReq)
                ? onebot_api_1.SendLikeReq.fromJSON(object.sendLikeReq)
                : undefined,
            setGroupKickReq: isSet(object.setGroupKickReq)
                ? onebot_api_1.SetGroupKickReq.fromJSON(object.setGroupKickReq)
                : undefined,
            setGroupBanReq: isSet(object.setGroupBanReq)
                ? onebot_api_1.SetGroupBanReq.fromJSON(object.setGroupBanReq)
                : undefined,
            setGroupAnonymousBanReq: isSet(object.setGroupAnonymousBanReq)
                ? onebot_api_1.SetGroupAnonymousBanReq.fromJSON(object.setGroupAnonymousBanReq)
                : undefined,
            setGroupWholeBanReq: isSet(object.setGroupWholeBanReq)
                ? onebot_api_1.SetGroupWholeBanReq.fromJSON(object.setGroupWholeBanReq)
                : undefined,
            setGroupAdminReq: isSet(object.setGroupAdminReq)
                ? onebot_api_1.SetGroupAdminReq.fromJSON(object.setGroupAdminReq)
                : undefined,
            setGroupAnonymousReq: isSet(object.setGroupAnonymousReq)
                ? onebot_api_1.SetGroupAnonymousReq.fromJSON(object.setGroupAnonymousReq)
                : undefined,
            setGroupCardReq: isSet(object.setGroupCardReq)
                ? onebot_api_1.SetGroupCardReq.fromJSON(object.setGroupCardReq)
                : undefined,
            setGroupNameReq: isSet(object.setGroupNameReq)
                ? onebot_api_1.SetGroupNameReq.fromJSON(object.setGroupNameReq)
                : undefined,
            setGroupLeaveReq: isSet(object.setGroupLeaveReq)
                ? onebot_api_1.SetGroupLeaveReq.fromJSON(object.setGroupLeaveReq)
                : undefined,
            setGroupSpecialTitleReq: isSet(object.setGroupSpecialTitleReq)
                ? onebot_api_1.SetGroupSpecialTitleReq.fromJSON(object.setGroupSpecialTitleReq)
                : undefined,
            setFriendAddRequestReq: isSet(object.setFriendAddRequestReq)
                ? onebot_api_1.SetFriendAddRequestReq.fromJSON(object.setFriendAddRequestReq)
                : undefined,
            setGroupAddRequestReq: isSet(object.setGroupAddRequestReq)
                ? onebot_api_1.SetGroupAddRequestReq.fromJSON(object.setGroupAddRequestReq)
                : undefined,
            getLoginInfoReq: isSet(object.getLoginInfoReq)
                ? onebot_api_1.GetLoginInfoReq.fromJSON(object.getLoginInfoReq)
                : undefined,
            getStrangerInfoReq: isSet(object.getStrangerInfoReq)
                ? onebot_api_1.GetStrangerInfoReq.fromJSON(object.getStrangerInfoReq)
                : undefined,
            getFriendListReq: isSet(object.getFriendListReq)
                ? onebot_api_1.GetFriendListReq.fromJSON(object.getFriendListReq)
                : undefined,
            getGroupInfoReq: isSet(object.getGroupInfoReq)
                ? onebot_api_1.GetGroupInfoReq.fromJSON(object.getGroupInfoReq)
                : undefined,
            getGroupListReq: isSet(object.getGroupListReq)
                ? onebot_api_1.GetGroupListReq.fromJSON(object.getGroupListReq)
                : undefined,
            getGroupMemberInfoReq: isSet(object.getGroupMemberInfoReq)
                ? onebot_api_1.GetGroupMemberInfoReq.fromJSON(object.getGroupMemberInfoReq)
                : undefined,
            getGroupMemberListReq: isSet(object.getGroupMemberListReq)
                ? onebot_api_1.GetGroupMemberListReq.fromJSON(object.getGroupMemberListReq)
                : undefined,
            getGroupHonorInfoReq: isSet(object.getGroupHonorInfoReq)
                ? onebot_api_1.GetGroupHonorInfoReq.fromJSON(object.getGroupHonorInfoReq)
                : undefined,
            getCookiesReq: isSet(object.getCookiesReq)
                ? onebot_api_1.GetCookiesReq.fromJSON(object.getCookiesReq)
                : undefined,
            getCsrfTokenReq: isSet(object.getCsrfTokenReq)
                ? onebot_api_1.GetCsrfTokenReq.fromJSON(object.getCsrfTokenReq)
                : undefined,
            getCredentialsReq: isSet(object.getCredentialsReq)
                ? onebot_api_1.GetCredentialsReq.fromJSON(object.getCredentialsReq)
                : undefined,
            getRecordReq: isSet(object.getRecordReq)
                ? onebot_api_1.GetRecordReq.fromJSON(object.getRecordReq)
                : undefined,
            getImageReq: isSet(object.getImageReq)
                ? onebot_api_1.GetImageReq.fromJSON(object.getImageReq)
                : undefined,
            canSendImageReq: isSet(object.canSendImageReq)
                ? onebot_api_1.CanSendImageReq.fromJSON(object.canSendImageReq)
                : undefined,
            canSendRecordReq: isSet(object.canSendRecordReq)
                ? onebot_api_1.CanSendRecordReq.fromJSON(object.canSendRecordReq)
                : undefined,
            getStatusReq: isSet(object.getStatusReq)
                ? onebot_api_1.GetStatusReq.fromJSON(object.getStatusReq)
                : undefined,
            getVersionInfoReq: isSet(object.getVersionInfoReq)
                ? onebot_api_1.GetVersionInfoReq.fromJSON(object.getVersionInfoReq)
                : undefined,
            setRestartReq: isSet(object.setRestartReq)
                ? onebot_api_1.SetRestartReq.fromJSON(object.setRestartReq)
                : undefined,
            cleanCacheReq: isSet(object.cleanCacheReq)
                ? onebot_api_1.CleanCacheReq.fromJSON(object.cleanCacheReq)
                : undefined,
            setGroupSignInReq: isSet(object.setGroupSignInReq)
                ? onebot_api_1.SetGroupSignInReq.fromJSON(object.setGroupSignInReq)
                : undefined,
            sendPrivateMsgResp: isSet(object.sendPrivateMsgResp)
                ? onebot_api_1.SendPrivateMsgResp.fromJSON(object.sendPrivateMsgResp)
                : undefined,
            sendGroupMsgResp: isSet(object.sendGroupMsgResp)
                ? onebot_api_1.SendGroupMsgResp.fromJSON(object.sendGroupMsgResp)
                : undefined,
            sendMsgResp: isSet(object.sendMsgResp)
                ? onebot_api_1.SendMsgResp.fromJSON(object.sendMsgResp)
                : undefined,
            deleteMsgResp: isSet(object.deleteMsgResp)
                ? onebot_api_1.DeleteMsgResp.fromJSON(object.deleteMsgResp)
                : undefined,
            getMsgResp: isSet(object.getMsgResp)
                ? onebot_api_1.GetMsgResp.fromJSON(object.getMsgResp)
                : undefined,
            getForwardMsgResp: isSet(object.getForwardMsgResp)
                ? onebot_api_1.GetForwardMsgResp.fromJSON(object.getForwardMsgResp)
                : undefined,
            sendLikeResp: isSet(object.sendLikeResp)
                ? onebot_api_1.SendLikeResp.fromJSON(object.sendLikeResp)
                : undefined,
            setGroupKickResp: isSet(object.setGroupKickResp)
                ? onebot_api_1.SetGroupKickResp.fromJSON(object.setGroupKickResp)
                : undefined,
            setGroupBanResp: isSet(object.setGroupBanResp)
                ? onebot_api_1.SetGroupBanResp.fromJSON(object.setGroupBanResp)
                : undefined,
            setGroupAnonymousBanResp: isSet(object.setGroupAnonymousBanResp)
                ? onebot_api_1.SetGroupAnonymousBanResp.fromJSON(object.setGroupAnonymousBanResp)
                : undefined,
            setGroupWholeBanResp: isSet(object.setGroupWholeBanResp)
                ? onebot_api_1.SetGroupWholeBanResp.fromJSON(object.setGroupWholeBanResp)
                : undefined,
            setGroupAdminResp: isSet(object.setGroupAdminResp)
                ? onebot_api_1.SetGroupAdminResp.fromJSON(object.setGroupAdminResp)
                : undefined,
            setGroupAnonymousResp: isSet(object.setGroupAnonymousResp)
                ? onebot_api_1.SetGroupAnonymousResp.fromJSON(object.setGroupAnonymousResp)
                : undefined,
            setGroupCardResp: isSet(object.setGroupCardResp)
                ? onebot_api_1.SetGroupCardResp.fromJSON(object.setGroupCardResp)
                : undefined,
            setGroupNameResp: isSet(object.setGroupNameResp)
                ? onebot_api_1.SetGroupNameResp.fromJSON(object.setGroupNameResp)
                : undefined,
            setGroupLeaveResp: isSet(object.setGroupLeaveResp)
                ? onebot_api_1.SetGroupLeaveResp.fromJSON(object.setGroupLeaveResp)
                : undefined,
            setGroupSpecialTitleResp: isSet(object.setGroupSpecialTitleResp)
                ? onebot_api_1.SetGroupSpecialTitleResp.fromJSON(object.setGroupSpecialTitleResp)
                : undefined,
            setFriendAddRequestResp: isSet(object.setFriendAddRequestResp)
                ? onebot_api_1.SetFriendAddRequestResp.fromJSON(object.setFriendAddRequestResp)
                : undefined,
            setGroupAddRequestResp: isSet(object.setGroupAddRequestResp)
                ? onebot_api_1.SetGroupAddRequestResp.fromJSON(object.setGroupAddRequestResp)
                : undefined,
            getLoginInfoResp: isSet(object.getLoginInfoResp)
                ? onebot_api_1.GetLoginInfoResp.fromJSON(object.getLoginInfoResp)
                : undefined,
            getStrangerInfoResp: isSet(object.getStrangerInfoResp)
                ? onebot_api_1.GetStrangerInfoResp.fromJSON(object.getStrangerInfoResp)
                : undefined,
            getFriendListResp: isSet(object.getFriendListResp)
                ? onebot_api_1.GetFriendListResp.fromJSON(object.getFriendListResp)
                : undefined,
            getGroupInfoResp: isSet(object.getGroupInfoResp)
                ? onebot_api_1.GetGroupInfoResp.fromJSON(object.getGroupInfoResp)
                : undefined,
            getGroupListResp: isSet(object.getGroupListResp)
                ? onebot_api_1.GetGroupListResp.fromJSON(object.getGroupListResp)
                : undefined,
            getGroupMemberInfoResp: isSet(object.getGroupMemberInfoResp)
                ? onebot_api_1.GetGroupMemberInfoResp.fromJSON(object.getGroupMemberInfoResp)
                : undefined,
            getGroupMemberListResp: isSet(object.getGroupMemberListResp)
                ? onebot_api_1.GetGroupMemberListResp.fromJSON(object.getGroupMemberListResp)
                : undefined,
            getGroupHonorInfoResp: isSet(object.getGroupHonorInfoResp)
                ? onebot_api_1.GetGroupHonorInfoResp.fromJSON(object.getGroupHonorInfoResp)
                : undefined,
            getCookiesResp: isSet(object.getCookiesResp)
                ? onebot_api_1.GetCookiesResp.fromJSON(object.getCookiesResp)
                : undefined,
            getCsrfTokenResp: isSet(object.getCsrfTokenResp)
                ? onebot_api_1.GetCsrfTokenResp.fromJSON(object.getCsrfTokenResp)
                : undefined,
            getCredentialsResp: isSet(object.getCredentialsResp)
                ? onebot_api_1.GetCredentialsResp.fromJSON(object.getCredentialsResp)
                : undefined,
            getRecordResp: isSet(object.getRecordResp)
                ? onebot_api_1.GetRecordResp.fromJSON(object.getRecordResp)
                : undefined,
            getImageResp: isSet(object.getImageResp)
                ? onebot_api_1.GetImageResp.fromJSON(object.getImageResp)
                : undefined,
            canSendImageResp: isSet(object.canSendImageResp)
                ? onebot_api_1.CanSendImageResp.fromJSON(object.canSendImageResp)
                : undefined,
            canSendRecordResp: isSet(object.canSendRecordResp)
                ? onebot_api_1.CanSendRecordResp.fromJSON(object.canSendRecordResp)
                : undefined,
            getStatusResp: isSet(object.getStatusResp)
                ? onebot_api_1.GetStatusResp.fromJSON(object.getStatusResp)
                : undefined,
            getVersionInfoResp: isSet(object.getVersionInfoResp)
                ? onebot_api_1.GetVersionInfoResp.fromJSON(object.getVersionInfoResp)
                : undefined,
            setRestartResp: isSet(object.setRestartResp)
                ? onebot_api_1.SetRestartResp.fromJSON(object.setRestartResp)
                : undefined,
            cleanCacheResp: isSet(object.cleanCacheResp)
                ? onebot_api_1.CleanCacheResp.fromJSON(object.cleanCacheResp)
                : undefined,
            setGroupSignInResp: isSet(object.setGroupSignInResp)
                ? onebot_api_1.SetGroupSignInResp.fromJSON(object.setGroupSignInResp)
                : undefined,
        };
    },
    toJSON: function (message) {
        var obj = {};
        message.botId !== undefined && (obj.botId = Math.round(message.botId));
        message.frameType !== undefined &&
            (obj.frameType = frame_FrameTypeToJSON(message.frameType));
        message.echo !== undefined && (obj.echo = message.echo);
        message.ok !== undefined && (obj.ok = message.ok);
        obj.extra = {};
        if (message.extra) {
            Object.entries(message.extra).forEach(function (_a) {
                var k = _a[0], v = _a[1];
                obj.extra[k] = v;
            });
        }
        message.privateMessageEvent !== undefined &&
            (obj.privateMessageEvent = message.privateMessageEvent
                ? onebot_event_1.PrivateMessageEvent.toJSON(message.privateMessageEvent)
                : undefined);
        message.groupMessageEvent !== undefined &&
            (obj.groupMessageEvent = message.groupMessageEvent
                ? onebot_event_1.GroupMessageEvent.toJSON(message.groupMessageEvent)
                : undefined);
        message.groupUploadNoticeEvent !== undefined &&
            (obj.groupUploadNoticeEvent = message.groupUploadNoticeEvent
                ? onebot_event_1.GroupUploadNoticeEvent.toJSON(message.groupUploadNoticeEvent)
                : undefined);
        message.groupAdminNoticeEvent !== undefined &&
            (obj.groupAdminNoticeEvent = message.groupAdminNoticeEvent
                ? onebot_event_1.GroupAdminNoticeEvent.toJSON(message.groupAdminNoticeEvent)
                : undefined);
        message.groupDecreaseNoticeEvent !== undefined &&
            (obj.groupDecreaseNoticeEvent = message.groupDecreaseNoticeEvent
                ? onebot_event_1.GroupDecreaseNoticeEvent.toJSON(message.groupDecreaseNoticeEvent)
                : undefined);
        message.groupIncreaseNoticeEvent !== undefined &&
            (obj.groupIncreaseNoticeEvent = message.groupIncreaseNoticeEvent
                ? onebot_event_1.GroupIncreaseNoticeEvent.toJSON(message.groupIncreaseNoticeEvent)
                : undefined);
        message.groupBanNoticeEvent !== undefined &&
            (obj.groupBanNoticeEvent = message.groupBanNoticeEvent
                ? onebot_event_1.GroupBanNoticeEvent.toJSON(message.groupBanNoticeEvent)
                : undefined);
        message.friendAddNoticeEvent !== undefined &&
            (obj.friendAddNoticeEvent = message.friendAddNoticeEvent
                ? onebot_event_1.FriendAddNoticeEvent.toJSON(message.friendAddNoticeEvent)
                : undefined);
        message.groupRecallNoticeEvent !== undefined &&
            (obj.groupRecallNoticeEvent = message.groupRecallNoticeEvent
                ? onebot_event_1.GroupRecallNoticeEvent.toJSON(message.groupRecallNoticeEvent)
                : undefined);
        message.friendRecallNoticeEvent !== undefined &&
            (obj.friendRecallNoticeEvent = message.friendRecallNoticeEvent
                ? onebot_event_1.FriendRecallNoticeEvent.toJSON(message.friendRecallNoticeEvent)
                : undefined);
        message.friendRequestEvent !== undefined &&
            (obj.friendRequestEvent = message.friendRequestEvent
                ? onebot_event_1.FriendRequestEvent.toJSON(message.friendRequestEvent)
                : undefined);
        message.groupRequestEvent !== undefined &&
            (obj.groupRequestEvent = message.groupRequestEvent
                ? onebot_event_1.GroupRequestEvent.toJSON(message.groupRequestEvent)
                : undefined);
        message.sendPrivateMsgReq !== undefined &&
            (obj.sendPrivateMsgReq = message.sendPrivateMsgReq
                ? onebot_api_1.SendPrivateMsgReq.toJSON(message.sendPrivateMsgReq)
                : undefined);
        message.sendGroupMsgReq !== undefined &&
            (obj.sendGroupMsgReq = message.sendGroupMsgReq
                ? onebot_api_1.SendGroupMsgReq.toJSON(message.sendGroupMsgReq)
                : undefined);
        message.sendMsgReq !== undefined &&
            (obj.sendMsgReq = message.sendMsgReq
                ? onebot_api_1.SendMsgReq.toJSON(message.sendMsgReq)
                : undefined);
        message.deleteMsgReq !== undefined &&
            (obj.deleteMsgReq = message.deleteMsgReq
                ? onebot_api_1.DeleteMsgReq.toJSON(message.deleteMsgReq)
                : undefined);
        message.getMsgReq !== undefined &&
            (obj.getMsgReq = message.getMsgReq
                ? onebot_api_1.GetMsgReq.toJSON(message.getMsgReq)
                : undefined);
        message.getForwardMsgReq !== undefined &&
            (obj.getForwardMsgReq = message.getForwardMsgReq
                ? onebot_api_1.GetForwardMsgReq.toJSON(message.getForwardMsgReq)
                : undefined);
        message.sendLikeReq !== undefined &&
            (obj.sendLikeReq = message.sendLikeReq
                ? onebot_api_1.SendLikeReq.toJSON(message.sendLikeReq)
                : undefined);
        message.setGroupKickReq !== undefined &&
            (obj.setGroupKickReq = message.setGroupKickReq
                ? onebot_api_1.SetGroupKickReq.toJSON(message.setGroupKickReq)
                : undefined);
        message.setGroupBanReq !== undefined &&
            (obj.setGroupBanReq = message.setGroupBanReq
                ? onebot_api_1.SetGroupBanReq.toJSON(message.setGroupBanReq)
                : undefined);
        message.setGroupAnonymousBanReq !== undefined &&
            (obj.setGroupAnonymousBanReq = message.setGroupAnonymousBanReq
                ? onebot_api_1.SetGroupAnonymousBanReq.toJSON(message.setGroupAnonymousBanReq)
                : undefined);
        message.setGroupWholeBanReq !== undefined &&
            (obj.setGroupWholeBanReq = message.setGroupWholeBanReq
                ? onebot_api_1.SetGroupWholeBanReq.toJSON(message.setGroupWholeBanReq)
                : undefined);
        message.setGroupAdminReq !== undefined &&
            (obj.setGroupAdminReq = message.setGroupAdminReq
                ? onebot_api_1.SetGroupAdminReq.toJSON(message.setGroupAdminReq)
                : undefined);
        message.setGroupAnonymousReq !== undefined &&
            (obj.setGroupAnonymousReq = message.setGroupAnonymousReq
                ? onebot_api_1.SetGroupAnonymousReq.toJSON(message.setGroupAnonymousReq)
                : undefined);
        message.setGroupCardReq !== undefined &&
            (obj.setGroupCardReq = message.setGroupCardReq
                ? onebot_api_1.SetGroupCardReq.toJSON(message.setGroupCardReq)
                : undefined);
        message.setGroupNameReq !== undefined &&
            (obj.setGroupNameReq = message.setGroupNameReq
                ? onebot_api_1.SetGroupNameReq.toJSON(message.setGroupNameReq)
                : undefined);
        message.setGroupLeaveReq !== undefined &&
            (obj.setGroupLeaveReq = message.setGroupLeaveReq
                ? onebot_api_1.SetGroupLeaveReq.toJSON(message.setGroupLeaveReq)
                : undefined);
        message.setGroupSpecialTitleReq !== undefined &&
            (obj.setGroupSpecialTitleReq = message.setGroupSpecialTitleReq
                ? onebot_api_1.SetGroupSpecialTitleReq.toJSON(message.setGroupSpecialTitleReq)
                : undefined);
        message.setFriendAddRequestReq !== undefined &&
            (obj.setFriendAddRequestReq = message.setFriendAddRequestReq
                ? onebot_api_1.SetFriendAddRequestReq.toJSON(message.setFriendAddRequestReq)
                : undefined);
        message.setGroupAddRequestReq !== undefined &&
            (obj.setGroupAddRequestReq = message.setGroupAddRequestReq
                ? onebot_api_1.SetGroupAddRequestReq.toJSON(message.setGroupAddRequestReq)
                : undefined);
        message.getLoginInfoReq !== undefined &&
            (obj.getLoginInfoReq = message.getLoginInfoReq
                ? onebot_api_1.GetLoginInfoReq.toJSON(message.getLoginInfoReq)
                : undefined);
        message.getStrangerInfoReq !== undefined &&
            (obj.getStrangerInfoReq = message.getStrangerInfoReq
                ? onebot_api_1.GetStrangerInfoReq.toJSON(message.getStrangerInfoReq)
                : undefined);
        message.getFriendListReq !== undefined &&
            (obj.getFriendListReq = message.getFriendListReq
                ? onebot_api_1.GetFriendListReq.toJSON(message.getFriendListReq)
                : undefined);
        message.getGroupInfoReq !== undefined &&
            (obj.getGroupInfoReq = message.getGroupInfoReq
                ? onebot_api_1.GetGroupInfoReq.toJSON(message.getGroupInfoReq)
                : undefined);
        message.getGroupListReq !== undefined &&
            (obj.getGroupListReq = message.getGroupListReq
                ? onebot_api_1.GetGroupListReq.toJSON(message.getGroupListReq)
                : undefined);
        message.getGroupMemberInfoReq !== undefined &&
            (obj.getGroupMemberInfoReq = message.getGroupMemberInfoReq
                ? onebot_api_1.GetGroupMemberInfoReq.toJSON(message.getGroupMemberInfoReq)
                : undefined);
        message.getGroupMemberListReq !== undefined &&
            (obj.getGroupMemberListReq = message.getGroupMemberListReq
                ? onebot_api_1.GetGroupMemberListReq.toJSON(message.getGroupMemberListReq)
                : undefined);
        message.getGroupHonorInfoReq !== undefined &&
            (obj.getGroupHonorInfoReq = message.getGroupHonorInfoReq
                ? onebot_api_1.GetGroupHonorInfoReq.toJSON(message.getGroupHonorInfoReq)
                : undefined);
        message.getCookiesReq !== undefined &&
            (obj.getCookiesReq = message.getCookiesReq
                ? onebot_api_1.GetCookiesReq.toJSON(message.getCookiesReq)
                : undefined);
        message.getCsrfTokenReq !== undefined &&
            (obj.getCsrfTokenReq = message.getCsrfTokenReq
                ? onebot_api_1.GetCsrfTokenReq.toJSON(message.getCsrfTokenReq)
                : undefined);
        message.getCredentialsReq !== undefined &&
            (obj.getCredentialsReq = message.getCredentialsReq
                ? onebot_api_1.GetCredentialsReq.toJSON(message.getCredentialsReq)
                : undefined);
        message.getRecordReq !== undefined &&
            (obj.getRecordReq = message.getRecordReq
                ? onebot_api_1.GetRecordReq.toJSON(message.getRecordReq)
                : undefined);
        message.getImageReq !== undefined &&
            (obj.getImageReq = message.getImageReq
                ? onebot_api_1.GetImageReq.toJSON(message.getImageReq)
                : undefined);
        message.canSendImageReq !== undefined &&
            (obj.canSendImageReq = message.canSendImageReq
                ? onebot_api_1.CanSendImageReq.toJSON(message.canSendImageReq)
                : undefined);
        message.canSendRecordReq !== undefined &&
            (obj.canSendRecordReq = message.canSendRecordReq
                ? onebot_api_1.CanSendRecordReq.toJSON(message.canSendRecordReq)
                : undefined);
        message.getStatusReq !== undefined &&
            (obj.getStatusReq = message.getStatusReq
                ? onebot_api_1.GetStatusReq.toJSON(message.getStatusReq)
                : undefined);
        message.getVersionInfoReq !== undefined &&
            (obj.getVersionInfoReq = message.getVersionInfoReq
                ? onebot_api_1.GetVersionInfoReq.toJSON(message.getVersionInfoReq)
                : undefined);
        message.setRestartReq !== undefined &&
            (obj.setRestartReq = message.setRestartReq
                ? onebot_api_1.SetRestartReq.toJSON(message.setRestartReq)
                : undefined);
        message.cleanCacheReq !== undefined &&
            (obj.cleanCacheReq = message.cleanCacheReq
                ? onebot_api_1.CleanCacheReq.toJSON(message.cleanCacheReq)
                : undefined);
        message.setGroupSignInReq !== undefined &&
            (obj.setGroupSignInReq = message.setGroupSignInReq
                ? onebot_api_1.SetGroupSignInReq.toJSON(message.setGroupSignInReq)
                : undefined);
        message.sendPrivateMsgResp !== undefined &&
            (obj.sendPrivateMsgResp = message.sendPrivateMsgResp
                ? onebot_api_1.SendPrivateMsgResp.toJSON(message.sendPrivateMsgResp)
                : undefined);
        message.sendGroupMsgResp !== undefined &&
            (obj.sendGroupMsgResp = message.sendGroupMsgResp
                ? onebot_api_1.SendGroupMsgResp.toJSON(message.sendGroupMsgResp)
                : undefined);
        message.sendMsgResp !== undefined &&
            (obj.sendMsgResp = message.sendMsgResp
                ? onebot_api_1.SendMsgResp.toJSON(message.sendMsgResp)
                : undefined);
        message.deleteMsgResp !== undefined &&
            (obj.deleteMsgResp = message.deleteMsgResp
                ? onebot_api_1.DeleteMsgResp.toJSON(message.deleteMsgResp)
                : undefined);
        message.getMsgResp !== undefined &&
            (obj.getMsgResp = message.getMsgResp
                ? onebot_api_1.GetMsgResp.toJSON(message.getMsgResp)
                : undefined);
        message.getForwardMsgResp !== undefined &&
            (obj.getForwardMsgResp = message.getForwardMsgResp
                ? onebot_api_1.GetForwardMsgResp.toJSON(message.getForwardMsgResp)
                : undefined);
        message.sendLikeResp !== undefined &&
            (obj.sendLikeResp = message.sendLikeResp
                ? onebot_api_1.SendLikeResp.toJSON(message.sendLikeResp)
                : undefined);
        message.setGroupKickResp !== undefined &&
            (obj.setGroupKickResp = message.setGroupKickResp
                ? onebot_api_1.SetGroupKickResp.toJSON(message.setGroupKickResp)
                : undefined);
        message.setGroupBanResp !== undefined &&
            (obj.setGroupBanResp = message.setGroupBanResp
                ? onebot_api_1.SetGroupBanResp.toJSON(message.setGroupBanResp)
                : undefined);
        message.setGroupAnonymousBanResp !== undefined &&
            (obj.setGroupAnonymousBanResp = message.setGroupAnonymousBanResp
                ? onebot_api_1.SetGroupAnonymousBanResp.toJSON(message.setGroupAnonymousBanResp)
                : undefined);
        message.setGroupWholeBanResp !== undefined &&
            (obj.setGroupWholeBanResp = message.setGroupWholeBanResp
                ? onebot_api_1.SetGroupWholeBanResp.toJSON(message.setGroupWholeBanResp)
                : undefined);
        message.setGroupAdminResp !== undefined &&
            (obj.setGroupAdminResp = message.setGroupAdminResp
                ? onebot_api_1.SetGroupAdminResp.toJSON(message.setGroupAdminResp)
                : undefined);
        message.setGroupAnonymousResp !== undefined &&
            (obj.setGroupAnonymousResp = message.setGroupAnonymousResp
                ? onebot_api_1.SetGroupAnonymousResp.toJSON(message.setGroupAnonymousResp)
                : undefined);
        message.setGroupCardResp !== undefined &&
            (obj.setGroupCardResp = message.setGroupCardResp
                ? onebot_api_1.SetGroupCardResp.toJSON(message.setGroupCardResp)
                : undefined);
        message.setGroupNameResp !== undefined &&
            (obj.setGroupNameResp = message.setGroupNameResp
                ? onebot_api_1.SetGroupNameResp.toJSON(message.setGroupNameResp)
                : undefined);
        message.setGroupLeaveResp !== undefined &&
            (obj.setGroupLeaveResp = message.setGroupLeaveResp
                ? onebot_api_1.SetGroupLeaveResp.toJSON(message.setGroupLeaveResp)
                : undefined);
        message.setGroupSpecialTitleResp !== undefined &&
            (obj.setGroupSpecialTitleResp = message.setGroupSpecialTitleResp
                ? onebot_api_1.SetGroupSpecialTitleResp.toJSON(message.setGroupSpecialTitleResp)
                : undefined);
        message.setFriendAddRequestResp !== undefined &&
            (obj.setFriendAddRequestResp = message.setFriendAddRequestResp
                ? onebot_api_1.SetFriendAddRequestResp.toJSON(message.setFriendAddRequestResp)
                : undefined);
        message.setGroupAddRequestResp !== undefined &&
            (obj.setGroupAddRequestResp = message.setGroupAddRequestResp
                ? onebot_api_1.SetGroupAddRequestResp.toJSON(message.setGroupAddRequestResp)
                : undefined);
        message.getLoginInfoResp !== undefined &&
            (obj.getLoginInfoResp = message.getLoginInfoResp
                ? onebot_api_1.GetLoginInfoResp.toJSON(message.getLoginInfoResp)
                : undefined);
        message.getStrangerInfoResp !== undefined &&
            (obj.getStrangerInfoResp = message.getStrangerInfoResp
                ? onebot_api_1.GetStrangerInfoResp.toJSON(message.getStrangerInfoResp)
                : undefined);
        message.getFriendListResp !== undefined &&
            (obj.getFriendListResp = message.getFriendListResp
                ? onebot_api_1.GetFriendListResp.toJSON(message.getFriendListResp)
                : undefined);
        message.getGroupInfoResp !== undefined &&
            (obj.getGroupInfoResp = message.getGroupInfoResp
                ? onebot_api_1.GetGroupInfoResp.toJSON(message.getGroupInfoResp)
                : undefined);
        message.getGroupListResp !== undefined &&
            (obj.getGroupListResp = message.getGroupListResp
                ? onebot_api_1.GetGroupListResp.toJSON(message.getGroupListResp)
                : undefined);
        message.getGroupMemberInfoResp !== undefined &&
            (obj.getGroupMemberInfoResp = message.getGroupMemberInfoResp
                ? onebot_api_1.GetGroupMemberInfoResp.toJSON(message.getGroupMemberInfoResp)
                : undefined);
        message.getGroupMemberListResp !== undefined &&
            (obj.getGroupMemberListResp = message.getGroupMemberListResp
                ? onebot_api_1.GetGroupMemberListResp.toJSON(message.getGroupMemberListResp)
                : undefined);
        message.getGroupHonorInfoResp !== undefined &&
            (obj.getGroupHonorInfoResp = message.getGroupHonorInfoResp
                ? onebot_api_1.GetGroupHonorInfoResp.toJSON(message.getGroupHonorInfoResp)
                : undefined);
        message.getCookiesResp !== undefined &&
            (obj.getCookiesResp = message.getCookiesResp
                ? onebot_api_1.GetCookiesResp.toJSON(message.getCookiesResp)
                : undefined);
        message.getCsrfTokenResp !== undefined &&
            (obj.getCsrfTokenResp = message.getCsrfTokenResp
                ? onebot_api_1.GetCsrfTokenResp.toJSON(message.getCsrfTokenResp)
                : undefined);
        message.getCredentialsResp !== undefined &&
            (obj.getCredentialsResp = message.getCredentialsResp
                ? onebot_api_1.GetCredentialsResp.toJSON(message.getCredentialsResp)
                : undefined);
        message.getRecordResp !== undefined &&
            (obj.getRecordResp = message.getRecordResp
                ? onebot_api_1.GetRecordResp.toJSON(message.getRecordResp)
                : undefined);
        message.getImageResp !== undefined &&
            (obj.getImageResp = message.getImageResp
                ? onebot_api_1.GetImageResp.toJSON(message.getImageResp)
                : undefined);
        message.canSendImageResp !== undefined &&
            (obj.canSendImageResp = message.canSendImageResp
                ? onebot_api_1.CanSendImageResp.toJSON(message.canSendImageResp)
                : undefined);
        message.canSendRecordResp !== undefined &&
            (obj.canSendRecordResp = message.canSendRecordResp
                ? onebot_api_1.CanSendRecordResp.toJSON(message.canSendRecordResp)
                : undefined);
        message.getStatusResp !== undefined &&
            (obj.getStatusResp = message.getStatusResp
                ? onebot_api_1.GetStatusResp.toJSON(message.getStatusResp)
                : undefined);
        message.getVersionInfoResp !== undefined &&
            (obj.getVersionInfoResp = message.getVersionInfoResp
                ? onebot_api_1.GetVersionInfoResp.toJSON(message.getVersionInfoResp)
                : undefined);
        message.setRestartResp !== undefined &&
            (obj.setRestartResp = message.setRestartResp
                ? onebot_api_1.SetRestartResp.toJSON(message.setRestartResp)
                : undefined);
        message.cleanCacheResp !== undefined &&
            (obj.cleanCacheResp = message.cleanCacheResp
                ? onebot_api_1.CleanCacheResp.toJSON(message.cleanCacheResp)
                : undefined);
        message.setGroupSignInResp !== undefined &&
            (obj.setGroupSignInResp = message.setGroupSignInResp
                ? onebot_api_1.SetGroupSignInResp.toJSON(message.setGroupSignInResp)
                : undefined);
        return obj;
    },
    fromPartial: function (object) {
        var _a, _b, _c, _d, _e;
        var message = createBaseFrame();
        message.botId = (_a = object.botId) !== null && _a !== void 0 ? _a : 0;
        message.frameType = (_b = object.frameType) !== null && _b !== void 0 ? _b : 0;
        message.echo = (_c = object.echo) !== null && _c !== void 0 ? _c : "";
        message.ok = (_d = object.ok) !== null && _d !== void 0 ? _d : false;
        message.extra = Object.entries((_e = object.extra) !== null && _e !== void 0 ? _e : {}).reduce(function (acc, _a) {
            var key = _a[0], value = _a[1];
            if (value !== undefined) {
                acc[key] = String(value);
            }
            return acc;
        }, {});
        message.privateMessageEvent =
            object.privateMessageEvent !== undefined &&
                object.privateMessageEvent !== null
                ? onebot_event_1.PrivateMessageEvent.fromPartial(object.privateMessageEvent)
                : undefined;
        message.groupMessageEvent =
            object.groupMessageEvent !== undefined &&
                object.groupMessageEvent !== null
                ? onebot_event_1.GroupMessageEvent.fromPartial(object.groupMessageEvent)
                : undefined;
        message.groupUploadNoticeEvent =
            object.groupUploadNoticeEvent !== undefined &&
                object.groupUploadNoticeEvent !== null
                ? onebot_event_1.GroupUploadNoticeEvent.fromPartial(object.groupUploadNoticeEvent)
                : undefined;
        message.groupAdminNoticeEvent =
            object.groupAdminNoticeEvent !== undefined &&
                object.groupAdminNoticeEvent !== null
                ? onebot_event_1.GroupAdminNoticeEvent.fromPartial(object.groupAdminNoticeEvent)
                : undefined;
        message.groupDecreaseNoticeEvent =
            object.groupDecreaseNoticeEvent !== undefined &&
                object.groupDecreaseNoticeEvent !== null
                ? onebot_event_1.GroupDecreaseNoticeEvent.fromPartial(object.groupDecreaseNoticeEvent)
                : undefined;
        message.groupIncreaseNoticeEvent =
            object.groupIncreaseNoticeEvent !== undefined &&
                object.groupIncreaseNoticeEvent !== null
                ? onebot_event_1.GroupIncreaseNoticeEvent.fromPartial(object.groupIncreaseNoticeEvent)
                : undefined;
        message.groupBanNoticeEvent =
            object.groupBanNoticeEvent !== undefined &&
                object.groupBanNoticeEvent !== null
                ? onebot_event_1.GroupBanNoticeEvent.fromPartial(object.groupBanNoticeEvent)
                : undefined;
        message.friendAddNoticeEvent =
            object.friendAddNoticeEvent !== undefined &&
                object.friendAddNoticeEvent !== null
                ? onebot_event_1.FriendAddNoticeEvent.fromPartial(object.friendAddNoticeEvent)
                : undefined;
        message.groupRecallNoticeEvent =
            object.groupRecallNoticeEvent !== undefined &&
                object.groupRecallNoticeEvent !== null
                ? onebot_event_1.GroupRecallNoticeEvent.fromPartial(object.groupRecallNoticeEvent)
                : undefined;
        message.friendRecallNoticeEvent =
            object.friendRecallNoticeEvent !== undefined &&
                object.friendRecallNoticeEvent !== null
                ? onebot_event_1.FriendRecallNoticeEvent.fromPartial(object.friendRecallNoticeEvent)
                : undefined;
        message.friendRequestEvent =
            object.friendRequestEvent !== undefined &&
                object.friendRequestEvent !== null
                ? onebot_event_1.FriendRequestEvent.fromPartial(object.friendRequestEvent)
                : undefined;
        message.groupRequestEvent =
            object.groupRequestEvent !== undefined &&
                object.groupRequestEvent !== null
                ? onebot_event_1.GroupRequestEvent.fromPartial(object.groupRequestEvent)
                : undefined;
        message.sendPrivateMsgReq =
            object.sendPrivateMsgReq !== undefined &&
                object.sendPrivateMsgReq !== null
                ? onebot_api_1.SendPrivateMsgReq.fromPartial(object.sendPrivateMsgReq)
                : undefined;
        message.sendGroupMsgReq =
            object.sendGroupMsgReq !== undefined && object.sendGroupMsgReq !== null
                ? onebot_api_1.SendGroupMsgReq.fromPartial(object.sendGroupMsgReq)
                : undefined;
        message.sendMsgReq =
            object.sendMsgReq !== undefined && object.sendMsgReq !== null
                ? onebot_api_1.SendMsgReq.fromPartial(object.sendMsgReq)
                : undefined;
        message.deleteMsgReq =
            object.deleteMsgReq !== undefined && object.deleteMsgReq !== null
                ? onebot_api_1.DeleteMsgReq.fromPartial(object.deleteMsgReq)
                : undefined;
        message.getMsgReq =
            object.getMsgReq !== undefined && object.getMsgReq !== null
                ? onebot_api_1.GetMsgReq.fromPartial(object.getMsgReq)
                : undefined;
        message.getForwardMsgReq =
            object.getForwardMsgReq !== undefined && object.getForwardMsgReq !== null
                ? onebot_api_1.GetForwardMsgReq.fromPartial(object.getForwardMsgReq)
                : undefined;
        message.sendLikeReq =
            object.sendLikeReq !== undefined && object.sendLikeReq !== null
                ? onebot_api_1.SendLikeReq.fromPartial(object.sendLikeReq)
                : undefined;
        message.setGroupKickReq =
            object.setGroupKickReq !== undefined && object.setGroupKickReq !== null
                ? onebot_api_1.SetGroupKickReq.fromPartial(object.setGroupKickReq)
                : undefined;
        message.setGroupBanReq =
            object.setGroupBanReq !== undefined && object.setGroupBanReq !== null
                ? onebot_api_1.SetGroupBanReq.fromPartial(object.setGroupBanReq)
                : undefined;
        message.setGroupAnonymousBanReq =
            object.setGroupAnonymousBanReq !== undefined &&
                object.setGroupAnonymousBanReq !== null
                ? onebot_api_1.SetGroupAnonymousBanReq.fromPartial(object.setGroupAnonymousBanReq)
                : undefined;
        message.setGroupWholeBanReq =
            object.setGroupWholeBanReq !== undefined &&
                object.setGroupWholeBanReq !== null
                ? onebot_api_1.SetGroupWholeBanReq.fromPartial(object.setGroupWholeBanReq)
                : undefined;
        message.setGroupAdminReq =
            object.setGroupAdminReq !== undefined && object.setGroupAdminReq !== null
                ? onebot_api_1.SetGroupAdminReq.fromPartial(object.setGroupAdminReq)
                : undefined;
        message.setGroupAnonymousReq =
            object.setGroupAnonymousReq !== undefined &&
                object.setGroupAnonymousReq !== null
                ? onebot_api_1.SetGroupAnonymousReq.fromPartial(object.setGroupAnonymousReq)
                : undefined;
        message.setGroupCardReq =
            object.setGroupCardReq !== undefined && object.setGroupCardReq !== null
                ? onebot_api_1.SetGroupCardReq.fromPartial(object.setGroupCardReq)
                : undefined;
        message.setGroupNameReq =
            object.setGroupNameReq !== undefined && object.setGroupNameReq !== null
                ? onebot_api_1.SetGroupNameReq.fromPartial(object.setGroupNameReq)
                : undefined;
        message.setGroupLeaveReq =
            object.setGroupLeaveReq !== undefined && object.setGroupLeaveReq !== null
                ? onebot_api_1.SetGroupLeaveReq.fromPartial(object.setGroupLeaveReq)
                : undefined;
        message.setGroupSpecialTitleReq =
            object.setGroupSpecialTitleReq !== undefined &&
                object.setGroupSpecialTitleReq !== null
                ? onebot_api_1.SetGroupSpecialTitleReq.fromPartial(object.setGroupSpecialTitleReq)
                : undefined;
        message.setFriendAddRequestReq =
            object.setFriendAddRequestReq !== undefined &&
                object.setFriendAddRequestReq !== null
                ? onebot_api_1.SetFriendAddRequestReq.fromPartial(object.setFriendAddRequestReq)
                : undefined;
        message.setGroupAddRequestReq =
            object.setGroupAddRequestReq !== undefined &&
                object.setGroupAddRequestReq !== null
                ? onebot_api_1.SetGroupAddRequestReq.fromPartial(object.setGroupAddRequestReq)
                : undefined;
        message.getLoginInfoReq =
            object.getLoginInfoReq !== undefined && object.getLoginInfoReq !== null
                ? onebot_api_1.GetLoginInfoReq.fromPartial(object.getLoginInfoReq)
                : undefined;
        message.getStrangerInfoReq =
            object.getStrangerInfoReq !== undefined &&
                object.getStrangerInfoReq !== null
                ? onebot_api_1.GetStrangerInfoReq.fromPartial(object.getStrangerInfoReq)
                : undefined;
        message.getFriendListReq =
            object.getFriendListReq !== undefined && object.getFriendListReq !== null
                ? onebot_api_1.GetFriendListReq.fromPartial(object.getFriendListReq)
                : undefined;
        message.getGroupInfoReq =
            object.getGroupInfoReq !== undefined && object.getGroupInfoReq !== null
                ? onebot_api_1.GetGroupInfoReq.fromPartial(object.getGroupInfoReq)
                : undefined;
        message.getGroupListReq =
            object.getGroupListReq !== undefined && object.getGroupListReq !== null
                ? onebot_api_1.GetGroupListReq.fromPartial(object.getGroupListReq)
                : undefined;
        message.getGroupMemberInfoReq =
            object.getGroupMemberInfoReq !== undefined &&
                object.getGroupMemberInfoReq !== null
                ? onebot_api_1.GetGroupMemberInfoReq.fromPartial(object.getGroupMemberInfoReq)
                : undefined;
        message.getGroupMemberListReq =
            object.getGroupMemberListReq !== undefined &&
                object.getGroupMemberListReq !== null
                ? onebot_api_1.GetGroupMemberListReq.fromPartial(object.getGroupMemberListReq)
                : undefined;
        message.getGroupHonorInfoReq =
            object.getGroupHonorInfoReq !== undefined &&
                object.getGroupHonorInfoReq !== null
                ? onebot_api_1.GetGroupHonorInfoReq.fromPartial(object.getGroupHonorInfoReq)
                : undefined;
        message.getCookiesReq =
            object.getCookiesReq !== undefined && object.getCookiesReq !== null
                ? onebot_api_1.GetCookiesReq.fromPartial(object.getCookiesReq)
                : undefined;
        message.getCsrfTokenReq =
            object.getCsrfTokenReq !== undefined && object.getCsrfTokenReq !== null
                ? onebot_api_1.GetCsrfTokenReq.fromPartial(object.getCsrfTokenReq)
                : undefined;
        message.getCredentialsReq =
            object.getCredentialsReq !== undefined &&
                object.getCredentialsReq !== null
                ? onebot_api_1.GetCredentialsReq.fromPartial(object.getCredentialsReq)
                : undefined;
        message.getRecordReq =
            object.getRecordReq !== undefined && object.getRecordReq !== null
                ? onebot_api_1.GetRecordReq.fromPartial(object.getRecordReq)
                : undefined;
        message.getImageReq =
            object.getImageReq !== undefined && object.getImageReq !== null
                ? onebot_api_1.GetImageReq.fromPartial(object.getImageReq)
                : undefined;
        message.canSendImageReq =
            object.canSendImageReq !== undefined && object.canSendImageReq !== null
                ? onebot_api_1.CanSendImageReq.fromPartial(object.canSendImageReq)
                : undefined;
        message.canSendRecordReq =
            object.canSendRecordReq !== undefined && object.canSendRecordReq !== null
                ? onebot_api_1.CanSendRecordReq.fromPartial(object.canSendRecordReq)
                : undefined;
        message.getStatusReq =
            object.getStatusReq !== undefined && object.getStatusReq !== null
                ? onebot_api_1.GetStatusReq.fromPartial(object.getStatusReq)
                : undefined;
        message.getVersionInfoReq =
            object.getVersionInfoReq !== undefined &&
                object.getVersionInfoReq !== null
                ? onebot_api_1.GetVersionInfoReq.fromPartial(object.getVersionInfoReq)
                : undefined;
        message.setRestartReq =
            object.setRestartReq !== undefined && object.setRestartReq !== null
                ? onebot_api_1.SetRestartReq.fromPartial(object.setRestartReq)
                : undefined;
        message.cleanCacheReq =
            object.cleanCacheReq !== undefined && object.cleanCacheReq !== null
                ? onebot_api_1.CleanCacheReq.fromPartial(object.cleanCacheReq)
                : undefined;
        message.setGroupSignInReq =
            object.setGroupSignInReq !== undefined &&
                object.setGroupSignInReq !== null
                ? onebot_api_1.SetGroupSignInReq.fromPartial(object.setGroupSignInReq)
                : undefined;
        message.sendPrivateMsgResp =
            object.sendPrivateMsgResp !== undefined &&
                object.sendPrivateMsgResp !== null
                ? onebot_api_1.SendPrivateMsgResp.fromPartial(object.sendPrivateMsgResp)
                : undefined;
        message.sendGroupMsgResp =
            object.sendGroupMsgResp !== undefined && object.sendGroupMsgResp !== null
                ? onebot_api_1.SendGroupMsgResp.fromPartial(object.sendGroupMsgResp)
                : undefined;
        message.sendMsgResp =
            object.sendMsgResp !== undefined && object.sendMsgResp !== null
                ? onebot_api_1.SendMsgResp.fromPartial(object.sendMsgResp)
                : undefined;
        message.deleteMsgResp =
            object.deleteMsgResp !== undefined && object.deleteMsgResp !== null
                ? onebot_api_1.DeleteMsgResp.fromPartial(object.deleteMsgResp)
                : undefined;
        message.getMsgResp =
            object.getMsgResp !== undefined && object.getMsgResp !== null
                ? onebot_api_1.GetMsgResp.fromPartial(object.getMsgResp)
                : undefined;
        message.getForwardMsgResp =
            object.getForwardMsgResp !== undefined &&
                object.getForwardMsgResp !== null
                ? onebot_api_1.GetForwardMsgResp.fromPartial(object.getForwardMsgResp)
                : undefined;
        message.sendLikeResp =
            object.sendLikeResp !== undefined && object.sendLikeResp !== null
                ? onebot_api_1.SendLikeResp.fromPartial(object.sendLikeResp)
                : undefined;
        message.setGroupKickResp =
            object.setGroupKickResp !== undefined && object.setGroupKickResp !== null
                ? onebot_api_1.SetGroupKickResp.fromPartial(object.setGroupKickResp)
                : undefined;
        message.setGroupBanResp =
            object.setGroupBanResp !== undefined && object.setGroupBanResp !== null
                ? onebot_api_1.SetGroupBanResp.fromPartial(object.setGroupBanResp)
                : undefined;
        message.setGroupAnonymousBanResp =
            object.setGroupAnonymousBanResp !== undefined &&
                object.setGroupAnonymousBanResp !== null
                ? onebot_api_1.SetGroupAnonymousBanResp.fromPartial(object.setGroupAnonymousBanResp)
                : undefined;
        message.setGroupWholeBanResp =
            object.setGroupWholeBanResp !== undefined &&
                object.setGroupWholeBanResp !== null
                ? onebot_api_1.SetGroupWholeBanResp.fromPartial(object.setGroupWholeBanResp)
                : undefined;
        message.setGroupAdminResp =
            object.setGroupAdminResp !== undefined &&
                object.setGroupAdminResp !== null
                ? onebot_api_1.SetGroupAdminResp.fromPartial(object.setGroupAdminResp)
                : undefined;
        message.setGroupAnonymousResp =
            object.setGroupAnonymousResp !== undefined &&
                object.setGroupAnonymousResp !== null
                ? onebot_api_1.SetGroupAnonymousResp.fromPartial(object.setGroupAnonymousResp)
                : undefined;
        message.setGroupCardResp =
            object.setGroupCardResp !== undefined && object.setGroupCardResp !== null
                ? onebot_api_1.SetGroupCardResp.fromPartial(object.setGroupCardResp)
                : undefined;
        message.setGroupNameResp =
            object.setGroupNameResp !== undefined && object.setGroupNameResp !== null
                ? onebot_api_1.SetGroupNameResp.fromPartial(object.setGroupNameResp)
                : undefined;
        message.setGroupLeaveResp =
            object.setGroupLeaveResp !== undefined &&
                object.setGroupLeaveResp !== null
                ? onebot_api_1.SetGroupLeaveResp.fromPartial(object.setGroupLeaveResp)
                : undefined;
        message.setGroupSpecialTitleResp =
            object.setGroupSpecialTitleResp !== undefined &&
                object.setGroupSpecialTitleResp !== null
                ? onebot_api_1.SetGroupSpecialTitleResp.fromPartial(object.setGroupSpecialTitleResp)
                : undefined;
        message.setFriendAddRequestResp =
            object.setFriendAddRequestResp !== undefined &&
                object.setFriendAddRequestResp !== null
                ? onebot_api_1.SetFriendAddRequestResp.fromPartial(object.setFriendAddRequestResp)
                : undefined;
        message.setGroupAddRequestResp =
            object.setGroupAddRequestResp !== undefined &&
                object.setGroupAddRequestResp !== null
                ? onebot_api_1.SetGroupAddRequestResp.fromPartial(object.setGroupAddRequestResp)
                : undefined;
        message.getLoginInfoResp =
            object.getLoginInfoResp !== undefined && object.getLoginInfoResp !== null
                ? onebot_api_1.GetLoginInfoResp.fromPartial(object.getLoginInfoResp)
                : undefined;
        message.getStrangerInfoResp =
            object.getStrangerInfoResp !== undefined &&
                object.getStrangerInfoResp !== null
                ? onebot_api_1.GetStrangerInfoResp.fromPartial(object.getStrangerInfoResp)
                : undefined;
        message.getFriendListResp =
            object.getFriendListResp !== undefined &&
                object.getFriendListResp !== null
                ? onebot_api_1.GetFriendListResp.fromPartial(object.getFriendListResp)
                : undefined;
        message.getGroupInfoResp =
            object.getGroupInfoResp !== undefined && object.getGroupInfoResp !== null
                ? onebot_api_1.GetGroupInfoResp.fromPartial(object.getGroupInfoResp)
                : undefined;
        message.getGroupListResp =
            object.getGroupListResp !== undefined && object.getGroupListResp !== null
                ? onebot_api_1.GetGroupListResp.fromPartial(object.getGroupListResp)
                : undefined;
        message.getGroupMemberInfoResp =
            object.getGroupMemberInfoResp !== undefined &&
                object.getGroupMemberInfoResp !== null
                ? onebot_api_1.GetGroupMemberInfoResp.fromPartial(object.getGroupMemberInfoResp)
                : undefined;
        message.getGroupMemberListResp =
            object.getGroupMemberListResp !== undefined &&
                object.getGroupMemberListResp !== null
                ? onebot_api_1.GetGroupMemberListResp.fromPartial(object.getGroupMemberListResp)
                : undefined;
        message.getGroupHonorInfoResp =
            object.getGroupHonorInfoResp !== undefined &&
                object.getGroupHonorInfoResp !== null
                ? onebot_api_1.GetGroupHonorInfoResp.fromPartial(object.getGroupHonorInfoResp)
                : undefined;
        message.getCookiesResp =
            object.getCookiesResp !== undefined && object.getCookiesResp !== null
                ? onebot_api_1.GetCookiesResp.fromPartial(object.getCookiesResp)
                : undefined;
        message.getCsrfTokenResp =
            object.getCsrfTokenResp !== undefined && object.getCsrfTokenResp !== null
                ? onebot_api_1.GetCsrfTokenResp.fromPartial(object.getCsrfTokenResp)
                : undefined;
        message.getCredentialsResp =
            object.getCredentialsResp !== undefined &&
                object.getCredentialsResp !== null
                ? onebot_api_1.GetCredentialsResp.fromPartial(object.getCredentialsResp)
                : undefined;
        message.getRecordResp =
            object.getRecordResp !== undefined && object.getRecordResp !== null
                ? onebot_api_1.GetRecordResp.fromPartial(object.getRecordResp)
                : undefined;
        message.getImageResp =
            object.getImageResp !== undefined && object.getImageResp !== null
                ? onebot_api_1.GetImageResp.fromPartial(object.getImageResp)
                : undefined;
        message.canSendImageResp =
            object.canSendImageResp !== undefined && object.canSendImageResp !== null
                ? onebot_api_1.CanSendImageResp.fromPartial(object.canSendImageResp)
                : undefined;
        message.canSendRecordResp =
            object.canSendRecordResp !== undefined &&
                object.canSendRecordResp !== null
                ? onebot_api_1.CanSendRecordResp.fromPartial(object.canSendRecordResp)
                : undefined;
        message.getStatusResp =
            object.getStatusResp !== undefined && object.getStatusResp !== null
                ? onebot_api_1.GetStatusResp.fromPartial(object.getStatusResp)
                : undefined;
        message.getVersionInfoResp =
            object.getVersionInfoResp !== undefined &&
                object.getVersionInfoResp !== null
                ? onebot_api_1.GetVersionInfoResp.fromPartial(object.getVersionInfoResp)
                : undefined;
        message.setRestartResp =
            object.setRestartResp !== undefined && object.setRestartResp !== null
                ? onebot_api_1.SetRestartResp.fromPartial(object.setRestartResp)
                : undefined;
        message.cleanCacheResp =
            object.cleanCacheResp !== undefined && object.cleanCacheResp !== null
                ? onebot_api_1.CleanCacheResp.fromPartial(object.cleanCacheResp)
                : undefined;
        message.setGroupSignInResp =
            object.setGroupSignInResp !== undefined &&
                object.setGroupSignInResp !== null
                ? onebot_api_1.SetGroupSignInResp.fromPartial(object.setGroupSignInResp)
                : undefined;
        return message;
    },
};
function createBaseFrame_ExtraEntry() {
    return { key: "", value: "" };
}
exports.Frame_ExtraEntry = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = _m0.Writer.create(); }
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== "") {
            writer.uint32(18).string(message.value);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseFrame_ExtraEntry();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.key = reader.string();
                    break;
                case 2:
                    message.value = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            key: isSet(object.key) ? String(object.key) : "",
            value: isSet(object.value) ? String(object.value) : "",
        };
    },
    toJSON: function (message) {
        var obj = {};
        message.key !== undefined && (obj.key = message.key);
        message.value !== undefined && (obj.value = message.value);
        return obj;
    },
    fromPartial: function (object) {
        var _a, _b;
        var message = createBaseFrame_ExtraEntry();
        message.key = (_a = object.key) !== null && _a !== void 0 ? _a : "";
        message.value = (_b = object.value) !== null && _b !== void 0 ? _b : "";
        return message;
    },
};
var globalThis = (function () {
    if (typeof globalThis !== "undefined")
        return globalThis;
    if (typeof self !== "undefined")
        return self;
    if (typeof window !== "undefined")
        return window;
    if (typeof global !== "undefined")
        return global;
    throw "Unable to locate global object";
})();
function longToNumber(long) {
    if (long.gt(Number.MAX_SAFE_INTEGER)) {
        throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
    }
    return long.toNumber();
}
if (_m0.util.Long !== long_1.default) {
    _m0.util.Long = long_1.default;
    _m0.configure();
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
